// @pulumi/aws/appconfig/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostedConfigurationVersion = exports.getEnvironmentsOutput = exports.getEnvironments = exports.getEnvironmentOutput = exports.getEnvironment = exports.getConfigurationProfilesOutput = exports.getConfigurationProfiles = exports.getConfigurationProfileOutput = exports.getConfigurationProfile = exports.ExtensionAssociation = exports.Extension = exports.EventIntegration = exports.Environment = exports.DeploymentStrategy = exports.Deployment = exports.ConfigurationProfile = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ConfigurationProfile = null;
utilities.lazyLoad(exports, ["ConfigurationProfile"], () => require("./configurationProfile"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.DeploymentStrategy = null;
utilities.lazyLoad(exports, ["DeploymentStrategy"], () => require("./deploymentStrategy"));
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.EventIntegration = null;
utilities.lazyLoad(exports, ["EventIntegration"], () => require("./eventIntegration"));
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.ExtensionAssociation = null;
utilities.lazyLoad(exports, ["ExtensionAssociation"], () => require("./extensionAssociation"));
exports.getConfigurationProfile = null;
exports.getConfigurationProfileOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfile", "getConfigurationProfileOutput"], () => require("./getConfigurationProfile"));
exports.getConfigurationProfiles = null;
exports.getConfigurationProfilesOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfiles", "getConfigurationProfilesOutput"], () => require("./getConfigurationProfiles"));
exports.getEnvironment = null;
exports.getEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getEnvironment", "getEnvironmentOutput"], () => require("./getEnvironment"));
exports.getEnvironments = null;
exports.getEnvironmentsOutput = null;
utilities.lazyLoad(exports, ["getEnvironments", "getEnvironmentsOutput"], () => require("./getEnvironments"));
exports.HostedConfigurationVersion = null;
utilities.lazyLoad(exports, ["HostedConfigurationVersion"], () => require("./hostedConfigurationVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:appconfig/application:Application":
                return new exports.Application(name, undefined, { urn });
            case "aws:appconfig/configurationProfile:ConfigurationProfile":
                return new exports.ConfigurationProfile(name, undefined, { urn });
            case "aws:appconfig/deployment:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "aws:appconfig/deploymentStrategy:DeploymentStrategy":
                return new exports.DeploymentStrategy(name, undefined, { urn });
            case "aws:appconfig/environment:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "aws:appconfig/eventIntegration:EventIntegration":
                return new exports.EventIntegration(name, undefined, { urn });
            case "aws:appconfig/extension:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "aws:appconfig/extensionAssociation:ExtensionAssociation":
                return new exports.ExtensionAssociation(name, undefined, { urn });
            case "aws:appconfig/hostedConfigurationVersion:HostedConfigurationVersion":
                return new exports.HostedConfigurationVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "appconfig/application", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/configurationProfile", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/deployment", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/deploymentStrategy", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/environment", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/eventIntegration", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/extension", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/extensionAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "appconfig/hostedConfigurationVersion", _module);
