// @pulumi/aws/opensearch/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutboundConnection = exports.InboundConnectionAccepter = exports.getDomainOutput = exports.getDomain = exports.DomainSamlOptions = exports.DomainPolicy = exports.Domain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Domain = null;
utilities.lazyLoad(exports, ["Domain"], () => require("./domain"));
exports.DomainPolicy = null;
utilities.lazyLoad(exports, ["DomainPolicy"], () => require("./domainPolicy"));
exports.DomainSamlOptions = null;
utilities.lazyLoad(exports, ["DomainSamlOptions"], () => require("./domainSamlOptions"));
exports.getDomain = null;
exports.getDomainOutput = null;
utilities.lazyLoad(exports, ["getDomain", "getDomainOutput"], () => require("./getDomain"));
exports.InboundConnectionAccepter = null;
utilities.lazyLoad(exports, ["InboundConnectionAccepter"], () => require("./inboundConnectionAccepter"));
exports.OutboundConnection = null;
utilities.lazyLoad(exports, ["OutboundConnection"], () => require("./outboundConnection"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:opensearch/domain:Domain":
                return new exports.Domain(name, undefined, { urn });
            case "aws:opensearch/domainPolicy:DomainPolicy":
                return new exports.DomainPolicy(name, undefined, { urn });
            case "aws:opensearch/domainSamlOptions:DomainSamlOptions":
                return new exports.DomainSamlOptions(name, undefined, { urn });
            case "aws:opensearch/inboundConnectionAccepter:InboundConnectionAccepter":
                return new exports.InboundConnectionAccepter(name, undefined, { urn });
            case "aws:opensearch/outboundConnection:OutboundConnection":
                return new exports.OutboundConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "opensearch/domain", _module);
pulumi.runtime.registerResourceModule("aws", "opensearch/domainPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "opensearch/domainSamlOptions", _module);
pulumi.runtime.registerResourceModule("aws", "opensearch/inboundConnectionAccepter", _module);
pulumi.runtime.registerResourceModule("aws", "opensearch/outboundConnection", _module);
