// @pulumi/aws/codebuild/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = exports.SourceCredential = exports.ResourcePolicy = exports.ReportGroup = exports.Project = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ReportGroup = null;
utilities.lazyLoad(exports, ["ReportGroup"], () => require("./reportGroup"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.SourceCredential = null;
utilities.lazyLoad(exports, ["SourceCredential"], () => require("./sourceCredential"));
exports.Webhook = null;
utilities.lazyLoad(exports, ["Webhook"], () => require("./webhook"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:codebuild/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws:codebuild/reportGroup:ReportGroup":
                return new exports.ReportGroup(name, undefined, { urn });
            case "aws:codebuild/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:codebuild/sourceCredential:SourceCredential":
                return new exports.SourceCredential(name, undefined, { urn });
            case "aws:codebuild/webhook:Webhook":
                return new exports.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "codebuild/project", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/reportGroup", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/sourceCredential", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/webhook", _module);
