// @pulumi/aws/elasticbeanstalk/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSolutionStackOutput = exports.getSolutionStack = exports.getHostedZoneOutput = exports.getHostedZone = exports.getApplicationOutput = exports.getApplication = exports.Environment = exports.ConfigurationTemplate = exports.ApplicationVersion = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ApplicationVersion = null;
utilities.lazyLoad(exports, ["ApplicationVersion"], () => require("./applicationVersion"));
exports.ConfigurationTemplate = null;
utilities.lazyLoad(exports, ["ConfigurationTemplate"], () => require("./configurationTemplate"));
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getHostedZone = null;
exports.getHostedZoneOutput = null;
utilities.lazyLoad(exports, ["getHostedZone", "getHostedZoneOutput"], () => require("./getHostedZone"));
exports.getSolutionStack = null;
exports.getSolutionStackOutput = null;
utilities.lazyLoad(exports, ["getSolutionStack", "getSolutionStackOutput"], () => require("./getSolutionStack"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:elasticbeanstalk/application:Application":
                return new exports.Application(name, undefined, { urn });
            case "aws:elasticbeanstalk/applicationVersion:ApplicationVersion":
                return new exports.ApplicationVersion(name, undefined, { urn });
            case "aws:elasticbeanstalk/configurationTemplate:ConfigurationTemplate":
                return new exports.ConfigurationTemplate(name, undefined, { urn });
            case "aws:elasticbeanstalk/environment:Environment":
                return new exports.Environment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "elasticbeanstalk/application", _module);
pulumi.runtime.registerResourceModule("aws", "elasticbeanstalk/applicationVersion", _module);
pulumi.runtime.registerResourceModule("aws", "elasticbeanstalk/configurationTemplate", _module);
pulumi.runtime.registerResourceModule("aws", "elasticbeanstalk/environment", _module);
