// @pulumi/aws/codecommit/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trigger = exports.Repository = exports.getRepositoryOutput = exports.getRepository = exports.getApprovalRuleTemplateOutput = exports.getApprovalRuleTemplate = exports.ApprovalRuleTemplateAssociation = exports.ApprovalRuleTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ApprovalRuleTemplate = null;
utilities.lazyLoad(exports, ["ApprovalRuleTemplate"], () => require("./approvalRuleTemplate"));
exports.ApprovalRuleTemplateAssociation = null;
utilities.lazyLoad(exports, ["ApprovalRuleTemplateAssociation"], () => require("./approvalRuleTemplateAssociation"));
exports.getApprovalRuleTemplate = null;
exports.getApprovalRuleTemplateOutput = null;
utilities.lazyLoad(exports, ["getApprovalRuleTemplate", "getApprovalRuleTemplateOutput"], () => require("./getApprovalRuleTemplate"));
exports.getRepository = null;
exports.getRepositoryOutput = null;
utilities.lazyLoad(exports, ["getRepository", "getRepositoryOutput"], () => require("./getRepository"));
exports.Repository = null;
utilities.lazyLoad(exports, ["Repository"], () => require("./repository"));
exports.Trigger = null;
utilities.lazyLoad(exports, ["Trigger"], () => require("./trigger"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:codecommit/approvalRuleTemplate:ApprovalRuleTemplate":
                return new exports.ApprovalRuleTemplate(name, undefined, { urn });
            case "aws:codecommit/approvalRuleTemplateAssociation:ApprovalRuleTemplateAssociation":
                return new exports.ApprovalRuleTemplateAssociation(name, undefined, { urn });
            case "aws:codecommit/repository:Repository":
                return new exports.Repository(name, undefined, { urn });
            case "aws:codecommit/trigger:Trigger":
                return new exports.Trigger(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "codecommit/approvalRuleTemplate", _module);
pulumi.runtime.registerResourceModule("aws", "codecommit/approvalRuleTemplateAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "codecommit/repository", _module);
pulumi.runtime.registerResourceModule("aws", "codecommit/trigger", _module);
