// @pulumi/aws/cloudformation/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StackSetInstance = exports.StackSet = exports.Stack = exports.getStackOutput = exports.getStack = exports.getExportOutput = exports.getExport = exports.getCloudFormationTypeOutput = exports.getCloudFormationType = exports.CloudFormationType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CloudFormationType = null;
utilities.lazyLoad(exports, ["CloudFormationType"], () => require("./cloudFormationType"));
exports.getCloudFormationType = null;
exports.getCloudFormationTypeOutput = null;
utilities.lazyLoad(exports, ["getCloudFormationType", "getCloudFormationTypeOutput"], () => require("./getCloudFormationType"));
exports.getExport = null;
exports.getExportOutput = null;
utilities.lazyLoad(exports, ["getExport", "getExportOutput"], () => require("./getExport"));
exports.getStack = null;
exports.getStackOutput = null;
utilities.lazyLoad(exports, ["getStack", "getStackOutput"], () => require("./getStack"));
exports.Stack = null;
utilities.lazyLoad(exports, ["Stack"], () => require("./stack"));
exports.StackSet = null;
utilities.lazyLoad(exports, ["StackSet"], () => require("./stackSet"));
exports.StackSetInstance = null;
utilities.lazyLoad(exports, ["StackSetInstance"], () => require("./stackSetInstance"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:cloudformation/cloudFormationType:CloudFormationType":
                return new exports.CloudFormationType(name, undefined, { urn });
            case "aws:cloudformation/stack:Stack":
                return new exports.Stack(name, undefined, { urn });
            case "aws:cloudformation/stackSet:StackSet":
                return new exports.StackSet(name, undefined, { urn });
            case "aws:cloudformation/stackSetInstance:StackSetInstance":
                return new exports.StackSetInstance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "cloudformation/cloudFormationType", _module);
pulumi.runtime.registerResourceModule("aws", "cloudformation/stack", _module);
pulumi.runtime.registerResourceModule("aws", "cloudformation/stackSet", _module);
pulumi.runtime.registerResourceModule("aws", "cloudformation/stackSetInstance", _module);
