// @pulumi/aws/ssmcontacts/contactChannel.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactChannel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS SSM Contacts Contact Channel.
 *
 * ## Example Usage
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ssmcontacts.ContactChannel("example", {
 *     contactId: "arn:aws:ssm-contacts:us-west-2:123456789012:contact/contactalias",
 *     deliveryAddress: {
 *         simpleAddress: "email@example.com",
 *     },
 *     type: "EMAIL",
 * });
 * ```
 * ### Usage with SSM Contact
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleContact = new aws.ssmcontacts.Contact("exampleContact", {
 *     alias: "example_contact",
 *     type: "PERSONAL",
 * });
 * const example = new aws.ssmcontacts.ContactChannel("example", {
 *     contactId: exampleContact.arn,
 *     deliveryAddress: {
 *         simpleAddress: "email@example.com",
 *     },
 *     type: "EMAIL",
 * });
 * ```
 *
 * ## Import
 *
 * Import SSM Contact Channel using the `ARN`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:ssmcontacts/contactChannel:ContactChannel example arn:aws:ssm-contacts:us-west-2:123456789012:contact-channel/example
 * ```
 */
class ContactChannel extends pulumi.CustomResource {
    /**
     * Get an existing ContactChannel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ContactChannel(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ContactChannel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ContactChannel.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["activationStatus"] = state ? state.activationStatus : undefined;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["contactId"] = state ? state.contactId : undefined;
            resourceInputs["deliveryAddress"] = state ? state.deliveryAddress : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.contactId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contactId'");
            }
            if ((!args || args.deliveryAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deliveryAddress'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["contactId"] = args ? args.contactId : undefined;
            resourceInputs["deliveryAddress"] = args ? args.deliveryAddress : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["activationStatus"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ContactChannel.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ContactChannel = ContactChannel;
/** @internal */
ContactChannel.__pulumiType = 'aws:ssmcontacts/contactChannel:ContactChannel';
