// @pulumi/aws/costexplorer/anomalySubscription.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnomalySubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a CE Anomaly Subscription.
 *
 * ## Example Usage
 * ### Basic Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const testAnomalyMonitor = new aws.costexplorer.AnomalyMonitor("testAnomalyMonitor", {
 *     monitorType: "DIMENSIONAL",
 *     monitorDimension: "SERVICE",
 * });
 * const testAnomalySubscription = new aws.costexplorer.AnomalySubscription("testAnomalySubscription", {
 *     threshold: 100,
 *     frequency: "DAILY",
 *     monitorArnLists: [testAnomalyMonitor.arn],
 *     subscribers: [{
 *         type: "EMAIL",
 *         address: "abc@example.com",
 *     }],
 * });
 * ```
 * ### Threshold Expression
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.costexplorer.AnomalySubscription("test", {
 *     frequency: "DAILY",
 *     monitorArnLists: [aws_ce_anomaly_monitor.test.arn],
 *     subscribers: [{
 *         type: "EMAIL",
 *         address: "abc@example.com",
 *     }],
 *     thresholdExpression: {
 *         dimension: {
 *             key: "ANOMALY_TOTAL_IMPACT_ABSOLUTE",
 *             values: ["100.0"],
 *             matchOptions: ["GREATER_THAN_OR_EQUAL"],
 *         },
 *     },
 * });
 * ```
 * ### SNS Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const costAnomalyUpdates = new aws.sns.Topic("costAnomalyUpdates", {});
 * const snsTopicPolicy = pulumi.all([costAnomalyUpdates.arn, costAnomalyUpdates.arn]).apply(([costAnomalyUpdatesArn, costAnomalyUpdatesArn1]) => aws.iam.getPolicyDocumentOutput({
 *     policyId: "__default_policy_ID",
 *     statements: [
 *         {
 *             sid: "AWSAnomalyDetectionSNSPublishingPermissions",
 *             actions: ["SNS:Publish"],
 *             effect: "Allow",
 *             principals: [{
 *                 type: "Service",
 *                 identifiers: ["costalerts.amazonaws.com"],
 *             }],
 *             resources: [costAnomalyUpdatesArn],
 *         },
 *         {
 *             sid: "__default_statement_ID",
 *             actions: [
 *                 "SNS:Subscribe",
 *                 "SNS:SetTopicAttributes",
 *                 "SNS:RemovePermission",
 *                 "SNS:Receive",
 *                 "SNS:Publish",
 *                 "SNS:ListSubscriptionsByTopic",
 *                 "SNS:GetTopicAttributes",
 *                 "SNS:DeleteTopic",
 *                 "SNS:AddPermission",
 *             ],
 *             conditions: [{
 *                 test: "StringEquals",
 *                 variable: "AWS:SourceOwner",
 *                 values: [_var["account-id"]],
 *             }],
 *             effect: "Allow",
 *             principals: [{
 *                 type: "AWS",
 *                 identifiers: ["*"],
 *             }],
 *             resources: [costAnomalyUpdatesArn1],
 *         },
 *     ],
 * }));
 * const _default = new aws.sns.TopicPolicy("default", {
 *     arn: costAnomalyUpdates.arn,
 *     policy: snsTopicPolicy.apply(snsTopicPolicy => snsTopicPolicy.json),
 * });
 * const anomalyMonitor = new aws.costexplorer.AnomalyMonitor("anomalyMonitor", {
 *     monitorType: "DIMENSIONAL",
 *     monitorDimension: "SERVICE",
 * });
 * const realtimeSubscription = new aws.costexplorer.AnomalySubscription("realtimeSubscription", {
 *     threshold: 0,
 *     frequency: "IMMEDIATE",
 *     monitorArnLists: [anomalyMonitor.arn],
 *     subscribers: [{
 *         type: "SNS",
 *         address: costAnomalyUpdates.arn,
 *     }],
 * }, {
 *     dependsOn: [_default],
 * });
 * ```
 *
 * ## Import
 *
 * `aws_ce_anomaly_subscription` can be imported using the `id`, e.g.
 *
 * ```sh
 *  $ pulumi import aws:costexplorer/anomalySubscription:AnomalySubscription example AnomalySubscriptionARN
 * ```
 */
class AnomalySubscription extends pulumi.CustomResource {
    /**
     * Get an existing AnomalySubscription resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AnomalySubscription(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AnomalySubscription.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnomalySubscription.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["frequency"] = state ? state.frequency : undefined;
            resourceInputs["monitorArnLists"] = state ? state.monitorArnLists : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["subscribers"] = state ? state.subscribers : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["threshold"] = state ? state.threshold : undefined;
            resourceInputs["thresholdExpression"] = state ? state.thresholdExpression : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.frequency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'frequency'");
            }
            if ((!args || args.monitorArnLists === undefined) && !opts.urn) {
                throw new Error("Missing required property 'monitorArnLists'");
            }
            if ((!args || args.subscribers === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subscribers'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["frequency"] = args ? args.frequency : undefined;
            resourceInputs["monitorArnLists"] = args ? args.monitorArnLists : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["subscribers"] = args ? args.subscribers : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["threshold"] = args ? args.threshold : undefined;
            resourceInputs["thresholdExpression"] = args ? args.thresholdExpression : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AnomalySubscription.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AnomalySubscription = AnomalySubscription;
/** @internal */
AnomalySubscription.__pulumiType = 'aws:costexplorer/anomalySubscription:AnomalySubscription';
