// @pulumi/aws/costexplorer/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getTagsOutput = exports.getTags = exports.getCostCategoryOutput = exports.getCostCategory = exports.CostCategory = exports.CostAllocationTag = exports.AnomalySubscription = exports.AnomalyMonitor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnomalyMonitor = null;
utilities.lazyLoad(exports, ["AnomalyMonitor"], () => require("./anomalyMonitor"));
exports.AnomalySubscription = null;
utilities.lazyLoad(exports, ["AnomalySubscription"], () => require("./anomalySubscription"));
exports.CostAllocationTag = null;
utilities.lazyLoad(exports, ["CostAllocationTag"], () => require("./costAllocationTag"));
exports.CostCategory = null;
utilities.lazyLoad(exports, ["CostCategory"], () => require("./costCategory"));
exports.getCostCategory = null;
exports.getCostCategoryOutput = null;
utilities.lazyLoad(exports, ["getCostCategory", "getCostCategoryOutput"], () => require("./getCostCategory"));
exports.getTags = null;
exports.getTagsOutput = null;
utilities.lazyLoad(exports, ["getTags", "getTagsOutput"], () => require("./getTags"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:costexplorer/anomalyMonitor:AnomalyMonitor":
                return new exports.AnomalyMonitor(name, undefined, { urn });
            case "aws:costexplorer/anomalySubscription:AnomalySubscription":
                return new exports.AnomalySubscription(name, undefined, { urn });
            case "aws:costexplorer/costAllocationTag:CostAllocationTag":
                return new exports.CostAllocationTag(name, undefined, { urn });
            case "aws:costexplorer/costCategory:CostCategory":
                return new exports.CostCategory(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "costexplorer/anomalyMonitor", _module);
pulumi.runtime.registerResourceModule("aws", "costexplorer/anomalySubscription", _module);
pulumi.runtime.registerResourceModule("aws", "costexplorer/costAllocationTag", _module);
pulumi.runtime.registerResourceModule("aws", "costexplorer/costCategory", _module);
