// @pulumi/mongodbatlas/cloudProviderAccessAuthorization.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudProviderAccessAuthorization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
class CloudProviderAccessAuthorization extends pulumi.CustomResource {
    /**
     * Get an existing CloudProviderAccessAuthorization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CloudProviderAccessAuthorization(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CloudProviderAccessAuthorization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CloudProviderAccessAuthorization.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["authorizedDate"] = state ? state.authorizedDate : undefined;
            resourceInputs["aws"] = state ? state.aws : undefined;
            resourceInputs["azure"] = state ? state.azure : undefined;
            resourceInputs["featureUsages"] = state ? state.featureUsages : undefined;
            resourceInputs["projectId"] = state ? state.projectId : undefined;
            resourceInputs["roleId"] = state ? state.roleId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.projectId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectId'");
            }
            if ((!args || args.roleId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleId'");
            }
            resourceInputs["aws"] = args ? args.aws : undefined;
            resourceInputs["azure"] = args ? args.azure : undefined;
            resourceInputs["projectId"] = args ? args.projectId : undefined;
            resourceInputs["roleId"] = args ? args.roleId : undefined;
            resourceInputs["authorizedDate"] = undefined /*out*/;
            resourceInputs["featureUsages"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CloudProviderAccessAuthorization.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CloudProviderAccessAuthorization = CloudProviderAccessAuthorization;
/** @internal */
CloudProviderAccessAuthorization.__pulumiType = 'mongodbatlas:index/cloudProviderAccessAuthorization:CloudProviderAccessAuthorization';
