// @pulumi/mongodbatlas/federatedSettingsIdentityProvider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FederatedSettingsIdentityProvider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * `mongodbatlas.FederatedSettingsIdentityProvider` provides an Atlas federated settings identity provider resource provides a subset of settings to be maintained post import of the existing resource.
 * ## Example Usage
 *
 * > **IMPORTANT** You **MUST** import this resource before you can manage it with this provider.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as mongodbatlas from "@pulumi/mongodbatlas";
 *
 * const identityProvider = new mongodbatlas.FederatedSettingsIdentityProvider("identityProvider", {
 *     associatedDomains: ["yourdomain.com"],
 *     federationSettingsId: "627a9687f7f7f7f774de306f14",
 *     issuerUri: "http://www.okta.com/exk17q7f7f7f7fp50h8",
 *     requestBinding: "HTTP-POST",
 *     responseSignatureAlgorithm: "SHA-256",
 *     ssoDebugEnabled: true,
 *     ssoUrl: "https://mysso.oktapreview.com/app/mysso_terraformtestsso/exk17q7f7f7f7f50h8/sso/saml",
 *     status: "ACTIVE",
 * });
 * ```
 *
 * ## Import
 *
 * Identity Provider __must__ be imported before using federation_settings_id-idp_id, e.g.
 *
 * ```sh
 *  $ pulumi import mongodbatlas:index/federatedSettingsIdentityProvider:FederatedSettingsIdentityProvider identity_provider 6287a663c660f52b1c441c6c-0oad4fas87jL5Xnk1297
 * ```
 *  For more information see[MongoDB Atlas API Reference.](https://www.mongodb.com/docs/atlas/reference/api/federation-configuration/)
 */
class FederatedSettingsIdentityProvider extends pulumi.CustomResource {
    /**
     * Get an existing FederatedSettingsIdentityProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new FederatedSettingsIdentityProvider(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FederatedSettingsIdentityProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FederatedSettingsIdentityProvider.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["associatedDomains"] = state ? state.associatedDomains : undefined;
            resourceInputs["federationSettingsId"] = state ? state.federationSettingsId : undefined;
            resourceInputs["issuerUri"] = state ? state.issuerUri : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["oktaIdpId"] = state ? state.oktaIdpId : undefined;
            resourceInputs["requestBinding"] = state ? state.requestBinding : undefined;
            resourceInputs["responseSignatureAlgorithm"] = state ? state.responseSignatureAlgorithm : undefined;
            resourceInputs["ssoDebugEnabled"] = state ? state.ssoDebugEnabled : undefined;
            resourceInputs["ssoUrl"] = state ? state.ssoUrl : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.federationSettingsId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'federationSettingsId'");
            }
            if ((!args || args.issuerUri === undefined) && !opts.urn) {
                throw new Error("Missing required property 'issuerUri'");
            }
            if ((!args || args.requestBinding === undefined) && !opts.urn) {
                throw new Error("Missing required property 'requestBinding'");
            }
            if ((!args || args.responseSignatureAlgorithm === undefined) && !opts.urn) {
                throw new Error("Missing required property 'responseSignatureAlgorithm'");
            }
            if ((!args || args.ssoDebugEnabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ssoDebugEnabled'");
            }
            if ((!args || args.ssoUrl === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ssoUrl'");
            }
            if ((!args || args.status === undefined) && !opts.urn) {
                throw new Error("Missing required property 'status'");
            }
            resourceInputs["associatedDomains"] = args ? args.associatedDomains : undefined;
            resourceInputs["federationSettingsId"] = args ? args.federationSettingsId : undefined;
            resourceInputs["issuerUri"] = args ? args.issuerUri : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["requestBinding"] = args ? args.requestBinding : undefined;
            resourceInputs["responseSignatureAlgorithm"] = args ? args.responseSignatureAlgorithm : undefined;
            resourceInputs["ssoDebugEnabled"] = args ? args.ssoDebugEnabled : undefined;
            resourceInputs["ssoUrl"] = args ? args.ssoUrl : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["oktaIdpId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(FederatedSettingsIdentityProvider.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FederatedSettingsIdentityProvider = FederatedSettingsIdentityProvider;
/** @internal */
FederatedSettingsIdentityProvider.__pulumiType = 'mongodbatlas:index/federatedSettingsIdentityProvider:FederatedSettingsIdentityProvider';
