// @pulumi/mongodbatlas/networkPeering.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkPeering = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * `mongodbatlas.NetworkPeering` provides a Network Peering Connection resource. The resource lets you create, edit and delete network peering connections. The resource requires your Project ID.
 *
 * Ensure you have first created a network container if it is required for your configuration.  See the networkContainer resource documentation to determine if you need a network container first.  Examples for creating both container and peering resource are shown below as well as examples for creating the peering connection only.
 *
 * > **GCP AND AZURE ONLY:** Connect via Peering Only mode is deprecated, so no longer needed.  See disable Peering Only mode to disable.
 *
 * > **AZURE ONLY:** To create the peering request with an Azure VNET, you must grant Atlas the following permissions on the virtual network.
 *     Microsoft.Network/virtualNetworks/virtualNetworkPeerings/read
 *     Microsoft.Network/virtualNetworks/virtualNetworkPeerings/write
 *     Microsoft.Network/virtualNetworks/virtualNetworkPeerings/delete
 *     Microsoft.Network/virtualNetworks/peer/action
 * For more information see https://docs.atlas.mongodb.com/security-vpc-peering/ and https://docs.atlas.mongodb.com/reference/api/vpc-create-peering-connection/
 *
 * > **Create a Whitelist:** Ensure you whitelist the private IP ranges of the subnets in which your application is hosted in order to connect to your Atlas cluster.  See the projectIpWhitelist resource.
 *
 * > **NOTE:** Groups and projects are synonymous terms. You may find **group_id** in the official documentation.
 *
 * ## Example Usage
 *
 * ### Container & Peering Connection
 * ### Example with AWS
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as mongodbatlas from "@pulumi/mongodbatlas";
 *
 * // Container example provided but not always required,
 * // see network_container documentation for details.
 * const testNetworkContainer = new mongodbatlas.NetworkContainer("testNetworkContainer", {
 *     projectId: local.project_id,
 *     atlasCidrBlock: "10.8.0.0/21",
 *     providerName: "AWS",
 *     regionName: "US_EAST_1",
 * });
 * // Create the peering connection request
 * const testNetworkPeering = new mongodbatlas.NetworkPeering("testNetworkPeering", {
 *     accepterRegionName: "us-east-1",
 *     projectId: local.project_id,
 *     containerId: "507f1f77bcf86cd799439011",
 *     providerName: "AWS",
 *     routeTableCidrBlock: "192.168.0.0/24",
 *     vpcId: "vpc-abc123abc123",
 *     awsAccountId: "abc123abc123",
 * });
 * // the following assumes an AWS provider is configured
 * // Accept the peering connection request
 * const peer = new aws.ec2.VpcPeeringConnectionAccepter("peer", {
 *     vpcPeeringConnectionId: testNetworkPeering.connectionId,
 *     autoAccept: true,
 * });
 * ```
 * ### Example with GCP
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as mongodbatlas from "@pulumi/mongodbatlas";
 *
 * // Container example provided but not always required,
 * // see network_container documentation for details.
 * const testNetworkContainer = new mongodbatlas.NetworkContainer("testNetworkContainer", {
 *     projectId: local.project_id,
 *     atlasCidrBlock: "10.8.0.0/21",
 *     providerName: "GCP",
 * });
 * // Create the peering connection request
 * const testNetworkPeering = new mongodbatlas.NetworkPeering("testNetworkPeering", {
 *     projectId: local.project_id,
 *     containerId: testNetworkContainer.containerId,
 *     providerName: "GCP",
 *     gcpProjectId: local.GCP_PROJECT_ID,
 *     networkName: "default",
 * });
 * const default = gcp.compute.getNetwork({
 *     name: "default",
 * });
 * // Create the GCP peer
 * const peering = new gcp.compute.NetworkPeering("peering", {
 *     network: _default.then(_default => _default.selfLink),
 *     peerNetwork: pulumi.interpolate`https://www.googleapis.com/compute/v1/projects/${testNetworkPeering.atlasGcpProjectId}/global/networks/${testNetworkPeering.atlasVpcName}`,
 * });
 * // Create the cluster once the peering connection is completed
 * const testCluster = new mongodbatlas.Cluster("testCluster", {
 *     projectId: local.project_id,
 *     numShards: 1,
 *     clusterType: "REPLICASET",
 *     replicationSpecs: [{
 *         numShards: 1,
 *         regionsConfigs: [{
 *             regionName: "US_EAST_4",
 *             electableNodes: 3,
 *             priority: 7,
 *             readOnlyNodes: 0,
 *         }],
 *     }],
 *     autoScalingDiskGbEnabled: true,
 *     mongoDbMajorVersion: "4.2",
 *     providerName: "GCP",
 *     providerInstanceSizeName: "M10",
 * }, {
 *     dependsOn: ["google_compute_network_peering.peering"],
 * });
 * ```
 * ### Example with Azure
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as mongodbatlas from "@pulumi/mongodbatlas";
 *
 * // Ensure you have created the required Azure service principal first, see
 * // see https://docs.atlas.mongodb.com/security-vpc-peering/
 * // Container example provided but not always required,
 * // see network_container documentation for details.
 * const testNetworkContainer = new mongodbatlas.NetworkContainer("testNetworkContainer", {
 *     projectId: local.project_id,
 *     atlasCidrBlock: local.ATLAS_CIDR_BLOCK,
 *     providerName: "AZURE",
 *     region: "US_EAST_2",
 * });
 * // Create the peering connection request
 * const testNetworkPeering = new mongodbatlas.NetworkPeering("testNetworkPeering", {
 *     projectId: local.project_id,
 *     containerId: testNetworkContainer.containerId,
 *     providerName: "AZURE",
 *     azureDirectoryId: local.AZURE_DIRECTORY_ID,
 *     azureSubscriptionId: local.AZURE_SUBSCRIPTION_ID,
 *     resourceGroupName: local.AZURE_RESOURCES_GROUP_NAME,
 *     vnetName: local.AZURE_VNET_NAME,
 * });
 * // Create the cluster once the peering connection is completed
 * const testCluster = new mongodbatlas.Cluster("testCluster", {
 *     projectId: local.project_id,
 *     clusterType: "REPLICASET",
 *     replicationSpecs: [{
 *         numShards: 1,
 *         regionsConfigs: [{
 *             regionName: "US_EAST_2",
 *             electableNodes: 3,
 *             priority: 7,
 *             readOnlyNodes: 0,
 *         }],
 *     }],
 *     autoScalingDiskGbEnabled: true,
 *     mongoDbMajorVersion: "4.2",
 *     providerName: "AZURE",
 *     providerDiskTypeName: "P4",
 *     providerInstanceSizeName: "M10",
 * }, {
 *     dependsOn: ["mongodbatlas_network_peering.test"],
 * });
 * ```
 * ### Peering Connection Only, Container Exists
 * You can create a peering connection if an appropriate container for your cloud provider already exists in your project (see the networkContainer resource for more information).  A container may already exist if you have already created a cluster in your project, if so you may obtain the `containerId` from the cluster resource as shown in the examples below.
 * ### Example with AWS
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as mongodbatlas from "@pulumi/mongodbatlas";
 *
 * // Create an Atlas cluster, this creates a container if one
 * // does not yet exist for this AWS region
 * const test = new mongodbatlas.Cluster("test", {
 *     projectId: local.project_id,
 *     clusterType: "REPLICASET",
 *     replicationSpecs: [{
 *         numShards: 1,
 *         regionsConfigs: [{
 *             regionName: "US_EAST_2",
 *             electableNodes: 3,
 *             priority: 7,
 *             readOnlyNodes: 0,
 *         }],
 *     }],
 *     autoScalingDiskGbEnabled: false,
 *     mongoDbMajorVersion: "4.2",
 *     providerName: "AWS",
 *     providerInstanceSizeName: "M10",
 * });
 * // the following assumes an AWS provider is configured
 * const _default = new aws.ec2.DefaultVpc("default", {tags: {
 *     Name: "Default VPC",
 * }});
 * // Create the peering connection request
 * const mongoPeer = new mongodbatlas.NetworkPeering("mongoPeer", {
 *     accepterRegionName: "us-east-2",
 *     projectId: local.project_id,
 *     containerId: test.containerId,
 *     providerName: "AWS",
 *     routeTableCidrBlock: "172.31.0.0/16",
 *     vpcId: _default.id,
 *     awsAccountId: local.AWS_ACCOUNT_ID,
 * });
 * // Accept the connection
 * const awsPeer = new aws.ec2.VpcPeeringConnectionAccepter("awsPeer", {
 *     vpcPeeringConnectionId: mongoPeer.connectionId,
 *     autoAccept: true,
 *     tags: {
 *         Side: "Accepter",
 *     },
 * });
 * ```
 * ### Example with GCP
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as mongodbatlas from "@pulumi/mongodbatlas";
 *
 * // Create an Atlas cluster, this creates a container if one
 * // does not yet exist for this GCP
 * const testCluster = new mongodbatlas.Cluster("testCluster", {
 *     projectId: local.project_id,
 *     clusterType: "REPLICASET",
 *     replicationSpecs: [{
 *         numShards: 1,
 *         regionsConfigs: [{
 *             regionName: "US_EAST_2",
 *             electableNodes: 3,
 *             priority: 7,
 *             readOnlyNodes: 0,
 *         }],
 *     }],
 *     autoScalingDiskGbEnabled: true,
 *     mongoDbMajorVersion: "4.2",
 *     providerName: "GCP",
 *     providerInstanceSizeName: "M10",
 * });
 * // Create the peering connection request
 * const testNetworkPeering = new mongodbatlas.NetworkPeering("testNetworkPeering", {
 *     projectId: local.project_id,
 *     atlasCidrBlock: "192.168.0.0/18",
 *     containerId: testCluster.containerId,
 *     providerName: "GCP",
 *     gcpProjectId: local.GCP_PROJECT_ID,
 *     networkName: "default",
 * });
 * const default = gcp.compute.getNetwork({
 *     name: "default",
 * });
 * // Create the GCP peer
 * const peering = new gcp.compute.NetworkPeering("peering", {
 *     network: _default.then(_default => _default.selfLink),
 *     peerNetwork: pulumi.interpolate`https://www.googleapis.com/compute/v1/projects/${testNetworkPeering.atlasGcpProjectId}/global/networks/${testNetworkPeering.atlasVpcName}`,
 * });
 * ```
 * ### Example with Azure
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as mongodbatlas from "@pulumi/mongodbatlas";
 *
 * // Ensure you have created the required Azure service principal first, see
 * // see https://docs.atlas.mongodb.com/security-vpc-peering/
 * // Create an Atlas cluster, this creates a container if one
 * // does not yet exist for this AZURE region
 * const testCluster = new mongodbatlas.Cluster("testCluster", {
 *     projectId: local.project_id,
 *     clusterType: "REPLICASET",
 *     replicationSpecs: [{
 *         numShards: 1,
 *         regionsConfigs: [{
 *             regionName: "US_EAST_2",
 *             electableNodes: 3,
 *             priority: 7,
 *             readOnlyNodes: 0,
 *         }],
 *     }],
 *     autoScalingDiskGbEnabled: false,
 *     mongoDbMajorVersion: "4.2",
 *     providerName: "AZURE",
 *     providerInstanceSizeName: "M10",
 * });
 * // Create the peering connection request
 * const testNetworkPeering = new mongodbatlas.NetworkPeering("testNetworkPeering", {
 *     projectId: local.project_id,
 *     containerId: testCluster.containerId,
 *     providerName: "AZURE",
 *     azureDirectoryId: local.AZURE_DIRECTORY_ID,
 *     azureSubscriptionId: local.AZURE_SUBSCRIPTION_ID,
 *     resourceGroupName: local.AZURE_RESOURCE_GROUP_NAME,
 *     vnetName: local.AZURE_VNET_NAME,
 * });
 * ```
 *
 * ## Import
 *
 * Network Peering Connections can be imported using project ID and network peering id, in the format `PROJECTID-PEERID-PROVIDERNAME`, e.g.
 *
 * ```sh
 *  $ pulumi import mongodbatlas:index/networkPeering:NetworkPeering my_peering 1112222b3bf99403840e8934-5cbf563d87d9d67253be590a-AWS
 * ```
 *  See detailed information for arguments and attributes[MongoDB API Network Peering Connection](https://docs.atlas.mongodb.com/reference/api/vpc-create-peering-connection/)
 *
 * -> __NOTE:__ If you need to get an existing container ID see the How-To Guide.
 */
class NetworkPeering extends pulumi.CustomResource {
    /**
     * Get an existing NetworkPeering resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NetworkPeering(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkPeering.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkPeering.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accepterRegionName"] = state ? state.accepterRegionName : undefined;
            resourceInputs["atlasCidrBlock"] = state ? state.atlasCidrBlock : undefined;
            resourceInputs["atlasGcpProjectId"] = state ? state.atlasGcpProjectId : undefined;
            resourceInputs["atlasId"] = state ? state.atlasId : undefined;
            resourceInputs["atlasVpcName"] = state ? state.atlasVpcName : undefined;
            resourceInputs["awsAccountId"] = state ? state.awsAccountId : undefined;
            resourceInputs["azureDirectoryId"] = state ? state.azureDirectoryId : undefined;
            resourceInputs["azureSubscriptionId"] = state ? state.azureSubscriptionId : undefined;
            resourceInputs["connectionId"] = state ? state.connectionId : undefined;
            resourceInputs["containerId"] = state ? state.containerId : undefined;
            resourceInputs["errorMessage"] = state ? state.errorMessage : undefined;
            resourceInputs["errorState"] = state ? state.errorState : undefined;
            resourceInputs["errorStateName"] = state ? state.errorStateName : undefined;
            resourceInputs["gcpProjectId"] = state ? state.gcpProjectId : undefined;
            resourceInputs["networkName"] = state ? state.networkName : undefined;
            resourceInputs["peerId"] = state ? state.peerId : undefined;
            resourceInputs["projectId"] = state ? state.projectId : undefined;
            resourceInputs["providerName"] = state ? state.providerName : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["routeTableCidrBlock"] = state ? state.routeTableCidrBlock : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["statusName"] = state ? state.statusName : undefined;
            resourceInputs["vnetName"] = state ? state.vnetName : undefined;
            resourceInputs["vpcId"] = state ? state.vpcId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.containerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerId'");
            }
            if ((!args || args.projectId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectId'");
            }
            if ((!args || args.providerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'providerName'");
            }
            resourceInputs["accepterRegionName"] = args ? args.accepterRegionName : undefined;
            resourceInputs["atlasCidrBlock"] = args ? args.atlasCidrBlock : undefined;
            resourceInputs["atlasGcpProjectId"] = args ? args.atlasGcpProjectId : undefined;
            resourceInputs["atlasVpcName"] = args ? args.atlasVpcName : undefined;
            resourceInputs["awsAccountId"] = args ? args.awsAccountId : undefined;
            resourceInputs["azureDirectoryId"] = args ? args.azureDirectoryId : undefined;
            resourceInputs["azureSubscriptionId"] = args ? args.azureSubscriptionId : undefined;
            resourceInputs["containerId"] = args ? args.containerId : undefined;
            resourceInputs["gcpProjectId"] = args ? args.gcpProjectId : undefined;
            resourceInputs["networkName"] = args ? args.networkName : undefined;
            resourceInputs["projectId"] = args ? args.projectId : undefined;
            resourceInputs["providerName"] = args ? args.providerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["routeTableCidrBlock"] = args ? args.routeTableCidrBlock : undefined;
            resourceInputs["vnetName"] = args ? args.vnetName : undefined;
            resourceInputs["vpcId"] = args ? args.vpcId : undefined;
            resourceInputs["atlasId"] = undefined /*out*/;
            resourceInputs["connectionId"] = undefined /*out*/;
            resourceInputs["errorMessage"] = undefined /*out*/;
            resourceInputs["errorState"] = undefined /*out*/;
            resourceInputs["errorStateName"] = undefined /*out*/;
            resourceInputs["peerId"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkPeering.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkPeering = NetworkPeering;
/** @internal */
NetworkPeering.__pulumiType = 'mongodbatlas:index/networkPeering:NetworkPeering';
