// @pulumi/mongodbatlas/config/vars.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const __config = new pulumi.Config("mongodbatlas");
Object.defineProperty(exports, "assumeRole", {
    get() {
        return __config.getObject("assumeRole");
    },
    enumerable: true,
});
Object.defineProperty(exports, "awsAccessKeyId", {
    get() {
        return __config.get("awsAccessKeyId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "awsSecretAccessKey", {
    get() {
        return __config.get("awsSecretAccessKey");
    },
    enumerable: true,
});
Object.defineProperty(exports, "awsSessionToken", {
    get() {
        return __config.get("awsSessionToken");
    },
    enumerable: true,
});
Object.defineProperty(exports, "baseUrl", {
    get() {
        return __config.get("baseUrl");
    },
    enumerable: true,
});
Object.defineProperty(exports, "isMongodbgovCloud", {
    get() {
        return __config.getObject("isMongodbgovCloud");
    },
    enumerable: true,
});
Object.defineProperty(exports, "privateKey", {
    get() {
        return __config.get("privateKey");
    },
    enumerable: true,
});
Object.defineProperty(exports, "publicKey", {
    get() {
        return __config.get("publicKey");
    },
    enumerable: true,
});
Object.defineProperty(exports, "realmBaseUrl", {
    get() {
        return __config.get("realmBaseUrl");
    },
    enumerable: true,
});
Object.defineProperty(exports, "region", {
    get() {
        return __config.get("region");
    },
    enumerable: true,
});
Object.defineProperty(exports, "secretName", {
    get() {
        return __config.get("secretName");
    },
    enumerable: true,
});
Object.defineProperty(exports, "stsEndpoint", {
    get() {
        return __config.get("stsEndpoint");
    },
    enumerable: true,
});
