// @pulumi/aws/schemas/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schema = exports.RegistryPolicy = exports.Registry = exports.Discoverer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Discoverer = null;
utilities.lazyLoad(exports, ["Discoverer"], () => require("./discoverer"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
exports.RegistryPolicy = null;
utilities.lazyLoad(exports, ["RegistryPolicy"], () => require("./registryPolicy"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:schemas/discoverer:Discoverer":
                return new exports.Discoverer(name, undefined, { urn });
            case "aws:schemas/registry:Registry":
                return new exports.Registry(name, undefined, { urn });
            case "aws:schemas/registryPolicy:RegistryPolicy":
                return new exports.RegistryPolicy(name, undefined, { urn });
            case "aws:schemas/schema:Schema":
                return new exports.Schema(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "schemas/discoverer", _module);
pulumi.runtime.registerResourceModule("aws", "schemas/registry", _module);
pulumi.runtime.registerResourceModule("aws", "schemas/registryPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "schemas/schema", _module);
