// @pulumi/awsx/types/enums/ec2/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-gen-awsx. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubnetType = exports.NatGatewayStrategy = void 0;
exports.NatGatewayStrategy = {
    /**
     * Do not create any NAT Gateways. Resources in private subnets will not be able to access the internet.
     */
    None: "None",
    /**
     * Create a single NAT Gateway for the entire VPC. This configuration is not recommended for production infrastructure as it creates a single point of failure.
     */
    Single: "Single",
    /**
     * Create a NAT Gateway in each availability zone. This is the recommended configuration for production infrastructure.
     */
    OnePerAz: "OnePerAz",
};
exports.SubnetType = {
    /**
     * A subnet whose hosts can directly communicate with the internet.
     */
    Public: "Public",
    /**
     * A subnet whose hosts can not directly communicate with the internet, but can initiate outbound network traffic via a NAT Gateway.
     */
    Private: "Private",
    /**
     * A subnet whose hosts have no connectivity with the internet.
     */
    Isolated: "Isolated",
};
