// @pulumi/aws/lex/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SlotType = exports.Intent = exports.getSlotTypeOutput = exports.getSlotType = exports.getIntentOutput = exports.getIntent = exports.getBotAliasOutput = exports.getBotAlias = exports.getBotOutput = exports.getBot = exports.BotAlias = exports.Bot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Bot = null;
utilities.lazyLoad(exports, ["Bot"], () => require("./bot"));
exports.BotAlias = null;
utilities.lazyLoad(exports, ["BotAlias"], () => require("./botAlias"));
exports.getBot = null;
exports.getBotOutput = null;
utilities.lazyLoad(exports, ["getBot", "getBotOutput"], () => require("./getBot"));
exports.getBotAlias = null;
exports.getBotAliasOutput = null;
utilities.lazyLoad(exports, ["getBotAlias", "getBotAliasOutput"], () => require("./getBotAlias"));
exports.getIntent = null;
exports.getIntentOutput = null;
utilities.lazyLoad(exports, ["getIntent", "getIntentOutput"], () => require("./getIntent"));
exports.getSlotType = null;
exports.getSlotTypeOutput = null;
utilities.lazyLoad(exports, ["getSlotType", "getSlotTypeOutput"], () => require("./getSlotType"));
exports.Intent = null;
utilities.lazyLoad(exports, ["Intent"], () => require("./intent"));
exports.SlotType = null;
utilities.lazyLoad(exports, ["SlotType"], () => require("./slotType"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:lex/bot:Bot":
                return new exports.Bot(name, undefined, { urn });
            case "aws:lex/botAlias:BotAlias":
                return new exports.BotAlias(name, undefined, { urn });
            case "aws:lex/intent:Intent":
                return new exports.Intent(name, undefined, { urn });
            case "aws:lex/slotType:SlotType":
                return new exports.SlotType(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "lex/bot", _module);
pulumi.runtime.registerResourceModule("aws", "lex/botAlias", _module);
pulumi.runtime.registerResourceModule("aws", "lex/intent", _module);
pulumi.runtime.registerResourceModule("aws", "lex/slotType", _module);
