// @pulumi/aws/cloudtrail/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trail = exports.getServiceAccountOutput = exports.getServiceAccount = exports.getFunctionOutput = exports.getFunction = exports.EventDataStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.EventDataStore = null;
utilities.lazyLoad(exports, ["EventDataStore"], () => require("./eventDataStore"));
exports.getFunction = null;
exports.getFunctionOutput = null;
utilities.lazyLoad(exports, ["getFunction", "getFunctionOutput"], () => require("./getFunction"));
exports.getServiceAccount = null;
exports.getServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getServiceAccount", "getServiceAccountOutput"], () => require("./getServiceAccount"));
exports.Trail = null;
utilities.lazyLoad(exports, ["Trail"], () => require("./trail"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:cloudtrail/eventDataStore:EventDataStore":
                return new exports.EventDataStore(name, undefined, { urn });
            case "aws:cloudtrail/trail:Trail":
                return new exports.Trail(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "cloudtrail/eventDataStore", _module);
pulumi.runtime.registerResourceModule("aws", "cloudtrail/trail", _module);
