// @pulumi/aws/athena/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workgroup = exports.NamedQuery = exports.Database = exports.DataCatalog = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataCatalog = null;
utilities.lazyLoad(exports, ["DataCatalog"], () => require("./dataCatalog"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.NamedQuery = null;
utilities.lazyLoad(exports, ["NamedQuery"], () => require("./namedQuery"));
exports.Workgroup = null;
utilities.lazyLoad(exports, ["Workgroup"], () => require("./workgroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:athena/dataCatalog:DataCatalog":
                return new exports.DataCatalog(name, undefined, { urn });
            case "aws:athena/database:Database":
                return new exports.Database(name, undefined, { urn });
            case "aws:athena/namedQuery:NamedQuery":
                return new exports.NamedQuery(name, undefined, { urn });
            case "aws:athena/workgroup:Workgroup":
                return new exports.Workgroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "athena/dataCatalog", _module);
pulumi.runtime.registerResourceModule("aws", "athena/database", _module);
pulumi.runtime.registerResourceModule("aws", "athena/namedQuery", _module);
pulumi.runtime.registerResourceModule("aws", "athena/workgroup", _module);
