// @pulumi/aws/neptune/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubnetGroup = exports.ParameterGroup = exports.GlobalCluster = exports.getOrderableDbInstanceOutput = exports.getOrderableDbInstance = exports.getEngineVersionOutput = exports.getEngineVersion = exports.EventSubscription = exports.ClusterSnapshot = exports.ClusterParameterGroup = exports.ClusterInstance = exports.ClusterEndpoint = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterEndpoint = null;
utilities.lazyLoad(exports, ["ClusterEndpoint"], () => require("./clusterEndpoint"));
exports.ClusterInstance = null;
utilities.lazyLoad(exports, ["ClusterInstance"], () => require("./clusterInstance"));
exports.ClusterParameterGroup = null;
utilities.lazyLoad(exports, ["ClusterParameterGroup"], () => require("./clusterParameterGroup"));
exports.ClusterSnapshot = null;
utilities.lazyLoad(exports, ["ClusterSnapshot"], () => require("./clusterSnapshot"));
exports.EventSubscription = null;
utilities.lazyLoad(exports, ["EventSubscription"], () => require("./eventSubscription"));
exports.getEngineVersion = null;
exports.getEngineVersionOutput = null;
utilities.lazyLoad(exports, ["getEngineVersion", "getEngineVersionOutput"], () => require("./getEngineVersion"));
exports.getOrderableDbInstance = null;
exports.getOrderableDbInstanceOutput = null;
utilities.lazyLoad(exports, ["getOrderableDbInstance", "getOrderableDbInstanceOutput"], () => require("./getOrderableDbInstance"));
exports.GlobalCluster = null;
utilities.lazyLoad(exports, ["GlobalCluster"], () => require("./globalCluster"));
exports.ParameterGroup = null;
utilities.lazyLoad(exports, ["ParameterGroup"], () => require("./parameterGroup"));
exports.SubnetGroup = null;
utilities.lazyLoad(exports, ["SubnetGroup"], () => require("./subnetGroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:neptune/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "aws:neptune/clusterEndpoint:ClusterEndpoint":
                return new exports.ClusterEndpoint(name, undefined, { urn });
            case "aws:neptune/clusterInstance:ClusterInstance":
                return new exports.ClusterInstance(name, undefined, { urn });
            case "aws:neptune/clusterParameterGroup:ClusterParameterGroup":
                return new exports.ClusterParameterGroup(name, undefined, { urn });
            case "aws:neptune/clusterSnapshot:ClusterSnapshot":
                return new exports.ClusterSnapshot(name, undefined, { urn });
            case "aws:neptune/eventSubscription:EventSubscription":
                return new exports.EventSubscription(name, undefined, { urn });
            case "aws:neptune/globalCluster:GlobalCluster":
                return new exports.GlobalCluster(name, undefined, { urn });
            case "aws:neptune/parameterGroup:ParameterGroup":
                return new exports.ParameterGroup(name, undefined, { urn });
            case "aws:neptune/subnetGroup:SubnetGroup":
                return new exports.SubnetGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "neptune/cluster", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/clusterEndpoint", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/clusterInstance", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/clusterParameterGroup", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/clusterSnapshot", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/eventSubscription", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/globalCluster", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/parameterGroup", _module);
pulumi.runtime.registerResourceModule("aws", "neptune/subnetGroup", _module);
