// @pulumi/aws/vpclattice/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetGroupAttachment = exports.TargetGroup = exports.ServiceNetworkVpcAssociation = exports.ServiceNetworkServiceAssociation = exports.ServiceNetwork = exports.Service = exports.ResourcePolicy = exports.ListenerRule = exports.Listener = exports.getServiceNetworkOutput = exports.getServiceNetwork = exports.getServiceOutput = exports.getService = exports.getListenerOutput = exports.getListener = exports.getAuthPolicyOutput = exports.getAuthPolicy = exports.AuthPolicy = exports.AccessLogSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessLogSubscription = null;
utilities.lazyLoad(exports, ["AccessLogSubscription"], () => require("./accessLogSubscription"));
exports.AuthPolicy = null;
utilities.lazyLoad(exports, ["AuthPolicy"], () => require("./authPolicy"));
exports.getAuthPolicy = null;
exports.getAuthPolicyOutput = null;
utilities.lazyLoad(exports, ["getAuthPolicy", "getAuthPolicyOutput"], () => require("./getAuthPolicy"));
exports.getListener = null;
exports.getListenerOutput = null;
utilities.lazyLoad(exports, ["getListener", "getListenerOutput"], () => require("./getListener"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceNetwork = null;
exports.getServiceNetworkOutput = null;
utilities.lazyLoad(exports, ["getServiceNetwork", "getServiceNetworkOutput"], () => require("./getServiceNetwork"));
exports.Listener = null;
utilities.lazyLoad(exports, ["Listener"], () => require("./listener"));
exports.ListenerRule = null;
utilities.lazyLoad(exports, ["ListenerRule"], () => require("./listenerRule"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceNetwork = null;
utilities.lazyLoad(exports, ["ServiceNetwork"], () => require("./serviceNetwork"));
exports.ServiceNetworkServiceAssociation = null;
utilities.lazyLoad(exports, ["ServiceNetworkServiceAssociation"], () => require("./serviceNetworkServiceAssociation"));
exports.ServiceNetworkVpcAssociation = null;
utilities.lazyLoad(exports, ["ServiceNetworkVpcAssociation"], () => require("./serviceNetworkVpcAssociation"));
exports.TargetGroup = null;
utilities.lazyLoad(exports, ["TargetGroup"], () => require("./targetGroup"));
exports.TargetGroupAttachment = null;
utilities.lazyLoad(exports, ["TargetGroupAttachment"], () => require("./targetGroupAttachment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:vpclattice/accessLogSubscription:AccessLogSubscription":
                return new exports.AccessLogSubscription(name, undefined, { urn });
            case "aws:vpclattice/authPolicy:AuthPolicy":
                return new exports.AuthPolicy(name, undefined, { urn });
            case "aws:vpclattice/listener:Listener":
                return new exports.Listener(name, undefined, { urn });
            case "aws:vpclattice/listenerRule:ListenerRule":
                return new exports.ListenerRule(name, undefined, { urn });
            case "aws:vpclattice/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:vpclattice/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "aws:vpclattice/serviceNetwork:ServiceNetwork":
                return new exports.ServiceNetwork(name, undefined, { urn });
            case "aws:vpclattice/serviceNetworkServiceAssociation:ServiceNetworkServiceAssociation":
                return new exports.ServiceNetworkServiceAssociation(name, undefined, { urn });
            case "aws:vpclattice/serviceNetworkVpcAssociation:ServiceNetworkVpcAssociation":
                return new exports.ServiceNetworkVpcAssociation(name, undefined, { urn });
            case "aws:vpclattice/targetGroup:TargetGroup":
                return new exports.TargetGroup(name, undefined, { urn });
            case "aws:vpclattice/targetGroupAttachment:TargetGroupAttachment":
                return new exports.TargetGroupAttachment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "vpclattice/accessLogSubscription", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/authPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/listener", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/listenerRule", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/service", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/serviceNetwork", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/serviceNetworkServiceAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/serviceNetworkVpcAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/targetGroup", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/targetGroupAttachment", _module);
