// @pulumi/aws/appsync/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.Resolver = exports.GraphQLApi = exports.Function = exports.DomainNameApiAssociation = exports.DomainName = exports.DataSource = exports.ApiKey = exports.ApiCache = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ApiCache = null;
utilities.lazyLoad(exports, ["ApiCache"], () => require("./apiCache"));
exports.ApiKey = null;
utilities.lazyLoad(exports, ["ApiKey"], () => require("./apiKey"));
exports.DataSource = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
exports.DomainName = null;
utilities.lazyLoad(exports, ["DomainName"], () => require("./domainName"));
exports.DomainNameApiAssociation = null;
utilities.lazyLoad(exports, ["DomainNameApiAssociation"], () => require("./domainNameApiAssociation"));
exports.Function = null;
utilities.lazyLoad(exports, ["Function"], () => require("./function"));
exports.GraphQLApi = null;
utilities.lazyLoad(exports, ["GraphQLApi"], () => require("./graphQLApi"));
exports.Resolver = null;
utilities.lazyLoad(exports, ["Resolver"], () => require("./resolver"));
exports.Type = null;
utilities.lazyLoad(exports, ["Type"], () => require("./type"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:appsync/apiCache:ApiCache":
                return new exports.ApiCache(name, undefined, { urn });
            case "aws:appsync/apiKey:ApiKey":
                return new exports.ApiKey(name, undefined, { urn });
            case "aws:appsync/dataSource:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "aws:appsync/domainName:DomainName":
                return new exports.DomainName(name, undefined, { urn });
            case "aws:appsync/domainNameApiAssociation:DomainNameApiAssociation":
                return new exports.DomainNameApiAssociation(name, undefined, { urn });
            case "aws:appsync/function:Function":
                return new exports.Function(name, undefined, { urn });
            case "aws:appsync/graphQLApi:GraphQLApi":
                return new exports.GraphQLApi(name, undefined, { urn });
            case "aws:appsync/resolver:Resolver":
                return new exports.Resolver(name, undefined, { urn });
            case "aws:appsync/type:Type":
                return new exports.Type(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "appsync/apiCache", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/apiKey", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/dataSource", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/domainName", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/domainNameApiAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/function", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/graphQLApi", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/resolver", _module);
pulumi.runtime.registerResourceModule("aws", "appsync/type", _module);
