// @pulumi/aws/securityhub/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandardsSubscription = exports.StandardsControl = exports.ProductSubscription = exports.OrganizationConfiguration = exports.OrganizationAdminAccount = exports.Member = exports.InviteAccepter = exports.Insight = exports.FindingAggregator = exports.ActionTarget = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.ActionTarget = null;
utilities.lazyLoad(exports, ["ActionTarget"], () => require("./actionTarget"));
exports.FindingAggregator = null;
utilities.lazyLoad(exports, ["FindingAggregator"], () => require("./findingAggregator"));
exports.Insight = null;
utilities.lazyLoad(exports, ["Insight"], () => require("./insight"));
exports.InviteAccepter = null;
utilities.lazyLoad(exports, ["InviteAccepter"], () => require("./inviteAccepter"));
exports.Member = null;
utilities.lazyLoad(exports, ["Member"], () => require("./member"));
exports.OrganizationAdminAccount = null;
utilities.lazyLoad(exports, ["OrganizationAdminAccount"], () => require("./organizationAdminAccount"));
exports.OrganizationConfiguration = null;
utilities.lazyLoad(exports, ["OrganizationConfiguration"], () => require("./organizationConfiguration"));
exports.ProductSubscription = null;
utilities.lazyLoad(exports, ["ProductSubscription"], () => require("./productSubscription"));
exports.StandardsControl = null;
utilities.lazyLoad(exports, ["StandardsControl"], () => require("./standardsControl"));
exports.StandardsSubscription = null;
utilities.lazyLoad(exports, ["StandardsSubscription"], () => require("./standardsSubscription"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:securityhub/account:Account":
                return new exports.Account(name, undefined, { urn });
            case "aws:securityhub/actionTarget:ActionTarget":
                return new exports.ActionTarget(name, undefined, { urn });
            case "aws:securityhub/findingAggregator:FindingAggregator":
                return new exports.FindingAggregator(name, undefined, { urn });
            case "aws:securityhub/insight:Insight":
                return new exports.Insight(name, undefined, { urn });
            case "aws:securityhub/inviteAccepter:InviteAccepter":
                return new exports.InviteAccepter(name, undefined, { urn });
            case "aws:securityhub/member:Member":
                return new exports.Member(name, undefined, { urn });
            case "aws:securityhub/organizationAdminAccount:OrganizationAdminAccount":
                return new exports.OrganizationAdminAccount(name, undefined, { urn });
            case "aws:securityhub/organizationConfiguration:OrganizationConfiguration":
                return new exports.OrganizationConfiguration(name, undefined, { urn });
            case "aws:securityhub/productSubscription:ProductSubscription":
                return new exports.ProductSubscription(name, undefined, { urn });
            case "aws:securityhub/standardsControl:StandardsControl":
                return new exports.StandardsControl(name, undefined, { urn });
            case "aws:securityhub/standardsSubscription:StandardsSubscription":
                return new exports.StandardsSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "securityhub/account", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/actionTarget", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/findingAggregator", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/insight", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/inviteAccepter", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/member", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/organizationAdminAccount", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/organizationConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/productSubscription", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/standardsControl", _module);
pulumi.runtime.registerResourceModule("aws", "securityhub/standardsSubscription", _module);
