// @pulumi/aws/apprunner/vpcIngressConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcIngressConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an App Runner VPC Ingress Connection.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.apprunner.VpcIngressConnection("example", {
 *     serviceArn: aws_apprunner_service.example.arn,
 *     ingressVpcConfiguration: {
 *         vpcId: aws_default_vpc["default"].id,
 *         vpcEndpointId: aws_vpc_endpoint.apprunner.id,
 *     },
 *     tags: {
 *         foo: "bar",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * App Runner VPC Ingress Connection can be imported by using the `arn`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:apprunner/vpcIngressConnection:VpcIngressConnection example "arn:aws:apprunner:us-west-2:837424938642:vpcingressconnection/example/b379f86381d74825832c2e82080342fa"
 * ```
 */
class VpcIngressConnection extends pulumi.CustomResource {
    /**
     * Get an existing VpcIngressConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcIngressConnection(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VpcIngressConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcIngressConnection.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["domainName"] = state ? state.domainName : undefined;
            resourceInputs["ingressVpcConfiguration"] = state ? state.ingressVpcConfiguration : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["serviceArn"] = state ? state.serviceArn : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.ingressVpcConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ingressVpcConfiguration'");
            }
            if ((!args || args.serviceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceArn'");
            }
            resourceInputs["ingressVpcConfiguration"] = args ? args.ingressVpcConfiguration : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["serviceArn"] = args ? args.serviceArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["domainName"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VpcIngressConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VpcIngressConnection = VpcIngressConnection;
/** @internal */
VpcIngressConnection.__pulumiType = 'aws:apprunner/vpcIngressConnection:VpcIngressConnection';
