// @pulumi/aws/organizations/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyAttachment = exports.Policy = exports.OrganizationalUnit = exports.Organization = exports.getResourceTagsOutput = exports.getResourceTags = exports.getPolicyOutput = exports.getPolicy = exports.getOrganizationalUnitsOutput = exports.getOrganizationalUnits = exports.getOrganizationalUnitDescendantAccountsOutput = exports.getOrganizationalUnitDescendantAccounts = exports.getOrganizationalUnitChildAccountsOutput = exports.getOrganizationalUnitChildAccounts = exports.getOrganization = exports.getDelegatedServicesOutput = exports.getDelegatedServices = exports.getDelegatedAdministratorsOutput = exports.getDelegatedAdministrators = exports.DelegatedAdministrator = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.DelegatedAdministrator = null;
utilities.lazyLoad(exports, ["DelegatedAdministrator"], () => require("./delegatedAdministrator"));
exports.getDelegatedAdministrators = null;
exports.getDelegatedAdministratorsOutput = null;
utilities.lazyLoad(exports, ["getDelegatedAdministrators", "getDelegatedAdministratorsOutput"], () => require("./getDelegatedAdministrators"));
exports.getDelegatedServices = null;
exports.getDelegatedServicesOutput = null;
utilities.lazyLoad(exports, ["getDelegatedServices", "getDelegatedServicesOutput"], () => require("./getDelegatedServices"));
exports.getOrganization = null;
utilities.lazyLoad(exports, ["getOrganization"], () => require("./getOrganization"));
exports.getOrganizationalUnitChildAccounts = null;
exports.getOrganizationalUnitChildAccountsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnitChildAccounts", "getOrganizationalUnitChildAccountsOutput"], () => require("./getOrganizationalUnitChildAccounts"));
exports.getOrganizationalUnitDescendantAccounts = null;
exports.getOrganizationalUnitDescendantAccountsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnitDescendantAccounts", "getOrganizationalUnitDescendantAccountsOutput"], () => require("./getOrganizationalUnitDescendantAccounts"));
exports.getOrganizationalUnits = null;
exports.getOrganizationalUnitsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnits", "getOrganizationalUnitsOutput"], () => require("./getOrganizationalUnits"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getResourceTags = null;
exports.getResourceTagsOutput = null;
utilities.lazyLoad(exports, ["getResourceTags", "getResourceTagsOutput"], () => require("./getResourceTags"));
exports.Organization = null;
utilities.lazyLoad(exports, ["Organization"], () => require("./organization"));
exports.OrganizationalUnit = null;
utilities.lazyLoad(exports, ["OrganizationalUnit"], () => require("./organizationalUnit"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.PolicyAttachment = null;
utilities.lazyLoad(exports, ["PolicyAttachment"], () => require("./policyAttachment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:organizations/account:Account":
                return new exports.Account(name, undefined, { urn });
            case "aws:organizations/delegatedAdministrator:DelegatedAdministrator":
                return new exports.DelegatedAdministrator(name, undefined, { urn });
            case "aws:organizations/organization:Organization":
                return new exports.Organization(name, undefined, { urn });
            case "aws:organizations/organizationalUnit:OrganizationalUnit":
                return new exports.OrganizationalUnit(name, undefined, { urn });
            case "aws:organizations/policy:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "aws:organizations/policyAttachment:PolicyAttachment":
                return new exports.PolicyAttachment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "organizations/account", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/delegatedAdministrator", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/organization", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/organizationalUnit", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/policy", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/policyAttachment", _module);
