// @pulumi/aws/waf/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.XssMatchSet = exports.WebAcl = exports.SqlInjectionMatchSet = exports.SizeConstraintSet = exports.RuleGroup = exports.Rule = exports.RegexPatternSet = exports.RegexMatchSet = exports.RateBasedRule = exports.IpSet = exports.getWebAclOutput = exports.getWebAcl = exports.getSubscribedRuleGroupOutput = exports.getSubscribedRuleGroup = exports.getRuleOutput = exports.getRule = exports.getRateBasedRuleOutput = exports.getRateBasedRule = exports.getIpsetOutput = exports.getIpset = exports.GeoMatchSet = exports.ByteMatchSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ByteMatchSet = null;
utilities.lazyLoad(exports, ["ByteMatchSet"], () => require("./byteMatchSet"));
exports.GeoMatchSet = null;
utilities.lazyLoad(exports, ["GeoMatchSet"], () => require("./geoMatchSet"));
exports.getIpset = null;
exports.getIpsetOutput = null;
utilities.lazyLoad(exports, ["getIpset", "getIpsetOutput"], () => require("./getIpset"));
exports.getRateBasedRule = null;
exports.getRateBasedRuleOutput = null;
utilities.lazyLoad(exports, ["getRateBasedRule", "getRateBasedRuleOutput"], () => require("./getRateBasedRule"));
exports.getRule = null;
exports.getRuleOutput = null;
utilities.lazyLoad(exports, ["getRule", "getRuleOutput"], () => require("./getRule"));
exports.getSubscribedRuleGroup = null;
exports.getSubscribedRuleGroupOutput = null;
utilities.lazyLoad(exports, ["getSubscribedRuleGroup", "getSubscribedRuleGroupOutput"], () => require("./getSubscribedRuleGroup"));
exports.getWebAcl = null;
exports.getWebAclOutput = null;
utilities.lazyLoad(exports, ["getWebAcl", "getWebAclOutput"], () => require("./getWebAcl"));
exports.IpSet = null;
utilities.lazyLoad(exports, ["IpSet"], () => require("./ipSet"));
exports.RateBasedRule = null;
utilities.lazyLoad(exports, ["RateBasedRule"], () => require("./rateBasedRule"));
exports.RegexMatchSet = null;
utilities.lazyLoad(exports, ["RegexMatchSet"], () => require("./regexMatchSet"));
exports.RegexPatternSet = null;
utilities.lazyLoad(exports, ["RegexPatternSet"], () => require("./regexPatternSet"));
exports.Rule = null;
utilities.lazyLoad(exports, ["Rule"], () => require("./rule"));
exports.RuleGroup = null;
utilities.lazyLoad(exports, ["RuleGroup"], () => require("./ruleGroup"));
exports.SizeConstraintSet = null;
utilities.lazyLoad(exports, ["SizeConstraintSet"], () => require("./sizeConstraintSet"));
exports.SqlInjectionMatchSet = null;
utilities.lazyLoad(exports, ["SqlInjectionMatchSet"], () => require("./sqlInjectionMatchSet"));
exports.WebAcl = null;
utilities.lazyLoad(exports, ["WebAcl"], () => require("./webAcl"));
exports.XssMatchSet = null;
utilities.lazyLoad(exports, ["XssMatchSet"], () => require("./xssMatchSet"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:waf/byteMatchSet:ByteMatchSet":
                return new exports.ByteMatchSet(name, undefined, { urn });
            case "aws:waf/geoMatchSet:GeoMatchSet":
                return new exports.GeoMatchSet(name, undefined, { urn });
            case "aws:waf/ipSet:IpSet":
                return new exports.IpSet(name, undefined, { urn });
            case "aws:waf/rateBasedRule:RateBasedRule":
                return new exports.RateBasedRule(name, undefined, { urn });
            case "aws:waf/regexMatchSet:RegexMatchSet":
                return new exports.RegexMatchSet(name, undefined, { urn });
            case "aws:waf/regexPatternSet:RegexPatternSet":
                return new exports.RegexPatternSet(name, undefined, { urn });
            case "aws:waf/rule:Rule":
                return new exports.Rule(name, undefined, { urn });
            case "aws:waf/ruleGroup:RuleGroup":
                return new exports.RuleGroup(name, undefined, { urn });
            case "aws:waf/sizeConstraintSet:SizeConstraintSet":
                return new exports.SizeConstraintSet(name, undefined, { urn });
            case "aws:waf/sqlInjectionMatchSet:SqlInjectionMatchSet":
                return new exports.SqlInjectionMatchSet(name, undefined, { urn });
            case "aws:waf/webAcl:WebAcl":
                return new exports.WebAcl(name, undefined, { urn });
            case "aws:waf/xssMatchSet:XssMatchSet":
                return new exports.XssMatchSet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "waf/byteMatchSet", _module);
pulumi.runtime.registerResourceModule("aws", "waf/geoMatchSet", _module);
pulumi.runtime.registerResourceModule("aws", "waf/ipSet", _module);
pulumi.runtime.registerResourceModule("aws", "waf/rateBasedRule", _module);
pulumi.runtime.registerResourceModule("aws", "waf/regexMatchSet", _module);
pulumi.runtime.registerResourceModule("aws", "waf/regexPatternSet", _module);
pulumi.runtime.registerResourceModule("aws", "waf/rule", _module);
pulumi.runtime.registerResourceModule("aws", "waf/ruleGroup", _module);
pulumi.runtime.registerResourceModule("aws", "waf/sizeConstraintSet", _module);
pulumi.runtime.registerResourceModule("aws", "waf/sqlInjectionMatchSet", _module);
pulumi.runtime.registerResourceModule("aws", "waf/webAcl", _module);
pulumi.runtime.registerResourceModule("aws", "waf/xssMatchSet", _module);
