// @pulumi/aws/rum/metricsDestination.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricsDestination = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a CloudWatch RUM Metrics Destination resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.rum.MetricsDestination("example", {
 *     appMonitorName: aws_rum_app_monitor.example.name,
 *     destination: "CloudWatch",
 * });
 * ```
 *
 * ## Import
 *
 * Cloudwatch RUM Metrics Destination can be imported using the `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:rum/metricsDestination:MetricsDestination example example
 * ```
 */
class MetricsDestination extends pulumi.CustomResource {
    /**
     * Get an existing MetricsDestination resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MetricsDestination(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MetricsDestination.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MetricsDestination.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appMonitorName"] = state ? state.appMonitorName : undefined;
            resourceInputs["destination"] = state ? state.destination : undefined;
            resourceInputs["destinationArn"] = state ? state.destinationArn : undefined;
            resourceInputs["iamRoleArn"] = state ? state.iamRoleArn : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.appMonitorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appMonitorName'");
            }
            if ((!args || args.destination === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destination'");
            }
            resourceInputs["appMonitorName"] = args ? args.appMonitorName : undefined;
            resourceInputs["destination"] = args ? args.destination : undefined;
            resourceInputs["destinationArn"] = args ? args.destinationArn : undefined;
            resourceInputs["iamRoleArn"] = args ? args.iamRoleArn : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MetricsDestination.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MetricsDestination = MetricsDestination;
/** @internal */
MetricsDestination.__pulumiType = 'aws:rum/metricsDestination:MetricsDestination';
