// @pulumi/aws/kendra/index_.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Index = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an Amazon Kendra Index resource.
 *
 * ## Example Usage
 * ### Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.kendra.Index("example", {
 *     description: "example",
 *     edition: "DEVELOPER_EDITION",
 *     roleArn: aws_iam_role["this"].arn,
 *     tags: {
 *         Key1: "Value1",
 *     },
 * });
 * ```
 * ### With capacity units
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.kendra.Index("example", {
 *     edition: "DEVELOPER_EDITION",
 *     roleArn: aws_iam_role["this"].arn,
 *     capacityUnits: {
 *         queryCapacityUnits: 2,
 *         storageCapacityUnits: 2,
 *     },
 * });
 * ```
 * ### With server side encryption configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.kendra.Index("example", {
 *     roleArn: aws_iam_role["this"].arn,
 *     serverSideEncryptionConfiguration: {
 *         kmsKeyId: data.aws_kms_key["this"].arn,
 *     },
 * });
 * ```
 * ### With Document Metadata Configuration Updates
 * ### Specifying the predefined elements
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.kendra.Index("example", {
 *     roleArn: aws_iam_role["this"].arn,
 *     documentMetadataConfigurationUpdates: [
 *         {
 *             name: "_authors",
 *             type: "STRING_LIST_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 importance: 1,
 *             },
 *         },
 *         {
 *             name: "_category",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_created_at",
 *             type: "DATE_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 freshness: false,
 *                 importance: 1,
 *                 duration: "25920000s",
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "_data_source_id",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_document_title",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: false,
 *                 searchable: true,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 2,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_excerpt_page_number",
 *             type: "LONG_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 importance: 2,
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "_faq_id",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_file_type",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_language_code",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_last_updated_at",
 *             type: "DATE_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 freshness: false,
 *                 importance: 1,
 *                 duration: "25920000s",
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "_source_uri",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_version",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_view_count",
 *             type: "LONG_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *     ],
 * });
 * ```
 * ### Appending additional elements
 *
 * The example below shows additional elements with names, `example-string-value`, `example-long-value`, `example-string-list-value`, `example-date-value` representing the 4 types of `STRING_VALUE`, `LONG_VALUE`, `STRING_LIST_VALUE`, `DATE_VALUE` respectively.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.kendra.Index("example", {
 *     roleArn: aws_iam_role["this"].arn,
 *     documentMetadataConfigurationUpdates: [
 *         {
 *             name: "_authors",
 *             type: "STRING_LIST_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 importance: 1,
 *             },
 *         },
 *         {
 *             name: "_category",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_created_at",
 *             type: "DATE_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 freshness: false,
 *                 importance: 1,
 *                 duration: "25920000s",
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "_data_source_id",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_document_title",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: false,
 *                 searchable: true,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 2,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_excerpt_page_number",
 *             type: "LONG_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 importance: 2,
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "_faq_id",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_file_type",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_language_code",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_last_updated_at",
 *             type: "DATE_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 freshness: false,
 *                 importance: 1,
 *                 duration: "25920000s",
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "_source_uri",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_version",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "_view_count",
 *             type: "LONG_VALUE",
 *             search: {
 *                 displayable: false,
 *                 facetable: false,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "example-string-value",
 *             type: "STRING_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: true,
 *                 searchable: true,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 valuesImportanceMap: {},
 *             },
 *         },
 *         {
 *             name: "example-long-value",
 *             type: "LONG_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: true,
 *                 searchable: false,
 *                 sortable: true,
 *             },
 *             relevance: {
 *                 importance: 1,
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *         {
 *             name: "example-string-list-value",
 *             type: "STRING_LIST_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: true,
 *                 searchable: true,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 importance: 1,
 *             },
 *         },
 *         {
 *             name: "example-date-value",
 *             type: "DATE_VALUE",
 *             search: {
 *                 displayable: true,
 *                 facetable: true,
 *                 searchable: false,
 *                 sortable: false,
 *             },
 *             relevance: {
 *                 freshness: false,
 *                 importance: 1,
 *                 duration: "25920000s",
 *                 rankOrder: "ASCENDING",
 *             },
 *         },
 *     ],
 * });
 * ```
 * ### With JSON token type configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.kendra.Index("example", {
 *     roleArn: aws_iam_role["this"].arn,
 *     userTokenConfigurations: {
 *         jsonTokenTypeConfiguration: {
 *             groupAttributeField: "groups",
 *             userNameAttributeField: "username",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Amazon Kendra Indexes can be imported using its `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:kendra/index:Index example 12345678-1234-5678-9123-123456789123
 * ```
 */
class Index extends pulumi.CustomResource {
    /**
     * Get an existing Index resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Index(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Index.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Index.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["capacityUnits"] = state ? state.capacityUnits : undefined;
            resourceInputs["createdAt"] = state ? state.createdAt : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["documentMetadataConfigurationUpdates"] = state ? state.documentMetadataConfigurationUpdates : undefined;
            resourceInputs["edition"] = state ? state.edition : undefined;
            resourceInputs["errorMessage"] = state ? state.errorMessage : undefined;
            resourceInputs["indexStatistics"] = state ? state.indexStatistics : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["roleArn"] = state ? state.roleArn : undefined;
            resourceInputs["serverSideEncryptionConfiguration"] = state ? state.serverSideEncryptionConfiguration : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["updatedAt"] = state ? state.updatedAt : undefined;
            resourceInputs["userContextPolicy"] = state ? state.userContextPolicy : undefined;
            resourceInputs["userGroupResolutionConfiguration"] = state ? state.userGroupResolutionConfiguration : undefined;
            resourceInputs["userTokenConfigurations"] = state ? state.userTokenConfigurations : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["capacityUnits"] = args ? args.capacityUnits : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["documentMetadataConfigurationUpdates"] = args ? args.documentMetadataConfigurationUpdates : undefined;
            resourceInputs["edition"] = args ? args.edition : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["serverSideEncryptionConfiguration"] = args ? args.serverSideEncryptionConfiguration : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["userContextPolicy"] = args ? args.userContextPolicy : undefined;
            resourceInputs["userGroupResolutionConfiguration"] = args ? args.userGroupResolutionConfiguration : undefined;
            resourceInputs["userTokenConfigurations"] = args ? args.userTokenConfigurations : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["errorMessage"] = undefined /*out*/;
            resourceInputs["indexStatistics"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Index.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Index = Index;
/** @internal */
Index.__pulumiType = 'aws:kendra/index:Index';
