// @pulumi/aws/kendra/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Thesaurus = exports.QuerySuggestionsBlockList = exports.Index = exports.getThesaurusOutput = exports.getThesaurus = exports.getQuerySuggestionsBlockListOutput = exports.getQuerySuggestionsBlockList = exports.getIndexOutput = exports.getIndex = exports.getFaqOutput = exports.getFaq = exports.getExperienceOutput = exports.getExperience = exports.Faq = exports.Experience = exports.DataSource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataSource = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
exports.Experience = null;
utilities.lazyLoad(exports, ["Experience"], () => require("./experience"));
exports.Faq = null;
utilities.lazyLoad(exports, ["Faq"], () => require("./faq"));
exports.getExperience = null;
exports.getExperienceOutput = null;
utilities.lazyLoad(exports, ["getExperience", "getExperienceOutput"], () => require("./getExperience"));
exports.getFaq = null;
exports.getFaqOutput = null;
utilities.lazyLoad(exports, ["getFaq", "getFaqOutput"], () => require("./getFaq"));
exports.getIndex = null;
exports.getIndexOutput = null;
utilities.lazyLoad(exports, ["getIndex", "getIndexOutput"], () => require("./getIndex"));
exports.getQuerySuggestionsBlockList = null;
exports.getQuerySuggestionsBlockListOutput = null;
utilities.lazyLoad(exports, ["getQuerySuggestionsBlockList", "getQuerySuggestionsBlockListOutput"], () => require("./getQuerySuggestionsBlockList"));
exports.getThesaurus = null;
exports.getThesaurusOutput = null;
utilities.lazyLoad(exports, ["getThesaurus", "getThesaurusOutput"], () => require("./getThesaurus"));
exports.Index = null;
utilities.lazyLoad(exports, ["Index"], () => require("./index_"));
exports.QuerySuggestionsBlockList = null;
utilities.lazyLoad(exports, ["QuerySuggestionsBlockList"], () => require("./querySuggestionsBlockList"));
exports.Thesaurus = null;
utilities.lazyLoad(exports, ["Thesaurus"], () => require("./thesaurus"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:kendra/dataSource:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "aws:kendra/experience:Experience":
                return new exports.Experience(name, undefined, { urn });
            case "aws:kendra/faq:Faq":
                return new exports.Faq(name, undefined, { urn });
            case "aws:kendra/index:Index":
                return new exports.Index(name, undefined, { urn });
            case "aws:kendra/querySuggestionsBlockList:QuerySuggestionsBlockList":
                return new exports.QuerySuggestionsBlockList(name, undefined, { urn });
            case "aws:kendra/thesaurus:Thesaurus":
                return new exports.Thesaurus(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "kendra/dataSource", _module);
pulumi.runtime.registerResourceModule("aws", "kendra/experience", _module);
pulumi.runtime.registerResourceModule("aws", "kendra/faq", _module);
pulumi.runtime.registerResourceModule("aws", "kendra/index", _module);
pulumi.runtime.registerResourceModule("aws", "kendra/querySuggestionsBlockList", _module);
pulumi.runtime.registerResourceModule("aws", "kendra/thesaurus", _module);
