// @pulumi/aws/ssoadmin/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PermissionsBoundaryAttachment = exports.PermissionSetInlinePolicy = exports.PermissionSet = exports.ManagedPolicyAttachment = exports.InstanceAccessControlAttributes = exports.getPermissionSetOutput = exports.getPermissionSet = exports.getInstances = exports.CustomerManagedPolicyAttachment = exports.AccountAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccountAssignment = null;
utilities.lazyLoad(exports, ["AccountAssignment"], () => require("./accountAssignment"));
exports.CustomerManagedPolicyAttachment = null;
utilities.lazyLoad(exports, ["CustomerManagedPolicyAttachment"], () => require("./customerManagedPolicyAttachment"));
exports.getInstances = null;
utilities.lazyLoad(exports, ["getInstances"], () => require("./getInstances"));
exports.getPermissionSet = null;
exports.getPermissionSetOutput = null;
utilities.lazyLoad(exports, ["getPermissionSet", "getPermissionSetOutput"], () => require("./getPermissionSet"));
exports.InstanceAccessControlAttributes = null;
utilities.lazyLoad(exports, ["InstanceAccessControlAttributes"], () => require("./instanceAccessControlAttributes"));
exports.ManagedPolicyAttachment = null;
utilities.lazyLoad(exports, ["ManagedPolicyAttachment"], () => require("./managedPolicyAttachment"));
exports.PermissionSet = null;
utilities.lazyLoad(exports, ["PermissionSet"], () => require("./permissionSet"));
exports.PermissionSetInlinePolicy = null;
utilities.lazyLoad(exports, ["PermissionSetInlinePolicy"], () => require("./permissionSetInlinePolicy"));
exports.PermissionsBoundaryAttachment = null;
utilities.lazyLoad(exports, ["PermissionsBoundaryAttachment"], () => require("./permissionsBoundaryAttachment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ssoadmin/accountAssignment:AccountAssignment":
                return new exports.AccountAssignment(name, undefined, { urn });
            case "aws:ssoadmin/customerManagedPolicyAttachment:CustomerManagedPolicyAttachment":
                return new exports.CustomerManagedPolicyAttachment(name, undefined, { urn });
            case "aws:ssoadmin/instanceAccessControlAttributes:InstanceAccessControlAttributes":
                return new exports.InstanceAccessControlAttributes(name, undefined, { urn });
            case "aws:ssoadmin/managedPolicyAttachment:ManagedPolicyAttachment":
                return new exports.ManagedPolicyAttachment(name, undefined, { urn });
            case "aws:ssoadmin/permissionSet:PermissionSet":
                return new exports.PermissionSet(name, undefined, { urn });
            case "aws:ssoadmin/permissionSetInlinePolicy:PermissionSetInlinePolicy":
                return new exports.PermissionSetInlinePolicy(name, undefined, { urn });
            case "aws:ssoadmin/permissionsBoundaryAttachment:PermissionsBoundaryAttachment":
                return new exports.PermissionsBoundaryAttachment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ssoadmin/accountAssignment", _module);
pulumi.runtime.registerResourceModule("aws", "ssoadmin/customerManagedPolicyAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "ssoadmin/instanceAccessControlAttributes", _module);
pulumi.runtime.registerResourceModule("aws", "ssoadmin/managedPolicyAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "ssoadmin/permissionSet", _module);
pulumi.runtime.registerResourceModule("aws", "ssoadmin/permissionSetInlinePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "ssoadmin/permissionsBoundaryAttachment", _module);
