// @pulumi/aws/dms/s3endpoint.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3Endpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a DMS (Data Migration Service) S3 endpoint resource. DMS S3 endpoints can be created, updated, deleted, and imported.
 *
 * > **Note:** AWS is deprecating `extraConnectionAttributes`, such as used with `aws.dms.Endpoint`. This resource is an alternative to `aws.dms.Endpoint` and does not use `extraConnectionAttributes`. (AWS currently includes `extraConnectionAttributes` in the raw responses to the AWS Provider requests and so they may be visible in the logs.)
 *
 * > **Note:** Some of this resource's arguments have default values that come from the AWS Provider. Other default values are provided by AWS and subject to change without notice. When relying on AWS defaults, the provider state will often have a zero value. For example, the AWS Provider does not provide a default for `cdcMaxBatchInterval` but the AWS default is `60` (seconds). However, the provider state will show `0` since this is the value return by AWS when no value is present. Below, we aim to flag the defaults that come from AWS (_e.g._, "AWS default...").
 *
 * ## Example Usage
 * ### Minimal Configuration
 *
 * This is the minimal configuration for an `aws.dms.S3Endpoint`. This endpoint will rely on the AWS Provider and AWS defaults.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.dms.S3Endpoint("example", {
 *     endpointId: "donnedtipi",
 *     endpointType: "target",
 *     bucketName: "beckut_name",
 *     serviceAccessRoleArn: aws_iam_role.example.arn,
 * }, {
 *     dependsOn: [aws_iam_role_policy.example],
 * });
 * ```
 * ### Complete Configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.dms.S3Endpoint("example", {
 *     endpointId: "donnedtipi",
 *     endpointType: "target",
 *     sslMode: "none",
 *     tags: {
 *         Name: "donnedtipi",
 *         Update: "to-update",
 *         Remove: "to-remove",
 *     },
 *     addColumnName: true,
 *     addTrailingPaddingCharacter: false,
 *     bucketFolder: "folder",
 *     bucketName: "bucket_name",
 *     cannedAclForObjects: "private",
 *     cdcInsertsAndUpdates: true,
 *     cdcInsertsOnly: false,
 *     cdcMaxBatchInterval: 100,
 *     cdcMinFileSize: 16,
 *     cdcPath: "cdc/path",
 *     compressionType: "GZIP",
 *     csvDelimiter: ";",
 *     csvNoSupValue: "x",
 *     csvNullValue: "?",
 *     csvRowDelimiter: "\\r\\n",
 *     dataFormat: "parquet",
 *     dataPageSize: 1100000,
 *     datePartitionDelimiter: "UNDERSCORE",
 *     datePartitionEnabled: true,
 *     datePartitionSequence: "yyyymmddhh",
 *     datePartitionTimezone: "Asia/Seoul",
 *     dictPageSizeLimit: 1000000,
 *     enableStatistics: false,
 *     encodingType: "plain",
 *     encryptionMode: "SSE_S3",
 *     expectedBucketOwner: data.aws_caller_identity.current.account_id,
 *     externalTableDefinition: "etd",
 *     ignoreHeaderRows: 1,
 *     includeOpForFullLoad: true,
 *     maxFileSize: 1000000,
 *     parquetTimestampInMillisecond: true,
 *     parquetVersion: "parquet-2-0",
 *     preserveTransactions: false,
 *     rfc4180: false,
 *     rowGroupLength: 11000,
 *     serverSideEncryptionKmsKeyId: aws_kms_key.example.arn,
 *     serviceAccessRoleArn: aws_iam_role.example.arn,
 *     timestampColumnName: "tx_commit_time",
 *     useCsvNoSupValue: false,
 *     useTaskStartTimeForFullLoadTimestamp: true,
 * }, {
 *     dependsOn: [aws_iam_role_policy.example],
 * });
 * ```
 *
 * ## Import
 *
 * Endpoints can be imported using the `endpoint_id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:dms/s3Endpoint:S3Endpoint example example-dms-endpoint-tf
 * ```
 */
class S3Endpoint extends pulumi.CustomResource {
    /**
     * Get an existing S3Endpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new S3Endpoint(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of S3Endpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === S3Endpoint.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["addColumnName"] = state ? state.addColumnName : undefined;
            resourceInputs["addTrailingPaddingCharacter"] = state ? state.addTrailingPaddingCharacter : undefined;
            resourceInputs["bucketFolder"] = state ? state.bucketFolder : undefined;
            resourceInputs["bucketName"] = state ? state.bucketName : undefined;
            resourceInputs["cannedAclForObjects"] = state ? state.cannedAclForObjects : undefined;
            resourceInputs["cdcInsertsAndUpdates"] = state ? state.cdcInsertsAndUpdates : undefined;
            resourceInputs["cdcInsertsOnly"] = state ? state.cdcInsertsOnly : undefined;
            resourceInputs["cdcMaxBatchInterval"] = state ? state.cdcMaxBatchInterval : undefined;
            resourceInputs["cdcMinFileSize"] = state ? state.cdcMinFileSize : undefined;
            resourceInputs["cdcPath"] = state ? state.cdcPath : undefined;
            resourceInputs["certificateArn"] = state ? state.certificateArn : undefined;
            resourceInputs["compressionType"] = state ? state.compressionType : undefined;
            resourceInputs["csvDelimiter"] = state ? state.csvDelimiter : undefined;
            resourceInputs["csvNoSupValue"] = state ? state.csvNoSupValue : undefined;
            resourceInputs["csvNullValue"] = state ? state.csvNullValue : undefined;
            resourceInputs["csvRowDelimiter"] = state ? state.csvRowDelimiter : undefined;
            resourceInputs["dataFormat"] = state ? state.dataFormat : undefined;
            resourceInputs["dataPageSize"] = state ? state.dataPageSize : undefined;
            resourceInputs["datePartitionDelimiter"] = state ? state.datePartitionDelimiter : undefined;
            resourceInputs["datePartitionEnabled"] = state ? state.datePartitionEnabled : undefined;
            resourceInputs["datePartitionSequence"] = state ? state.datePartitionSequence : undefined;
            resourceInputs["datePartitionTimezone"] = state ? state.datePartitionTimezone : undefined;
            resourceInputs["detachTargetOnLobLookupFailureParquet"] = state ? state.detachTargetOnLobLookupFailureParquet : undefined;
            resourceInputs["dictPageSizeLimit"] = state ? state.dictPageSizeLimit : undefined;
            resourceInputs["enableStatistics"] = state ? state.enableStatistics : undefined;
            resourceInputs["encodingType"] = state ? state.encodingType : undefined;
            resourceInputs["encryptionMode"] = state ? state.encryptionMode : undefined;
            resourceInputs["endpointArn"] = state ? state.endpointArn : undefined;
            resourceInputs["endpointId"] = state ? state.endpointId : undefined;
            resourceInputs["endpointType"] = state ? state.endpointType : undefined;
            resourceInputs["engineDisplayName"] = state ? state.engineDisplayName : undefined;
            resourceInputs["expectedBucketOwner"] = state ? state.expectedBucketOwner : undefined;
            resourceInputs["externalId"] = state ? state.externalId : undefined;
            resourceInputs["externalTableDefinition"] = state ? state.externalTableDefinition : undefined;
            resourceInputs["ignoreHeaderRows"] = state ? state.ignoreHeaderRows : undefined;
            resourceInputs["includeOpForFullLoad"] = state ? state.includeOpForFullLoad : undefined;
            resourceInputs["kmsKeyArn"] = state ? state.kmsKeyArn : undefined;
            resourceInputs["maxFileSize"] = state ? state.maxFileSize : undefined;
            resourceInputs["parquetTimestampInMillisecond"] = state ? state.parquetTimestampInMillisecond : undefined;
            resourceInputs["parquetVersion"] = state ? state.parquetVersion : undefined;
            resourceInputs["preserveTransactions"] = state ? state.preserveTransactions : undefined;
            resourceInputs["rfc4180"] = state ? state.rfc4180 : undefined;
            resourceInputs["rowGroupLength"] = state ? state.rowGroupLength : undefined;
            resourceInputs["serverSideEncryptionKmsKeyId"] = state ? state.serverSideEncryptionKmsKeyId : undefined;
            resourceInputs["serviceAccessRoleArn"] = state ? state.serviceAccessRoleArn : undefined;
            resourceInputs["sslMode"] = state ? state.sslMode : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["timestampColumnName"] = state ? state.timestampColumnName : undefined;
            resourceInputs["useCsvNoSupValue"] = state ? state.useCsvNoSupValue : undefined;
            resourceInputs["useTaskStartTimeForFullLoadTimestamp"] = state ? state.useTaskStartTimeForFullLoadTimestamp : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.bucketName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'bucketName'");
            }
            if ((!args || args.endpointId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointId'");
            }
            if ((!args || args.endpointType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointType'");
            }
            if ((!args || args.serviceAccessRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceAccessRoleArn'");
            }
            resourceInputs["addColumnName"] = args ? args.addColumnName : undefined;
            resourceInputs["addTrailingPaddingCharacter"] = args ? args.addTrailingPaddingCharacter : undefined;
            resourceInputs["bucketFolder"] = args ? args.bucketFolder : undefined;
            resourceInputs["bucketName"] = args ? args.bucketName : undefined;
            resourceInputs["cannedAclForObjects"] = args ? args.cannedAclForObjects : undefined;
            resourceInputs["cdcInsertsAndUpdates"] = args ? args.cdcInsertsAndUpdates : undefined;
            resourceInputs["cdcInsertsOnly"] = args ? args.cdcInsertsOnly : undefined;
            resourceInputs["cdcMaxBatchInterval"] = args ? args.cdcMaxBatchInterval : undefined;
            resourceInputs["cdcMinFileSize"] = args ? args.cdcMinFileSize : undefined;
            resourceInputs["cdcPath"] = args ? args.cdcPath : undefined;
            resourceInputs["certificateArn"] = args ? args.certificateArn : undefined;
            resourceInputs["compressionType"] = args ? args.compressionType : undefined;
            resourceInputs["csvDelimiter"] = args ? args.csvDelimiter : undefined;
            resourceInputs["csvNoSupValue"] = args ? args.csvNoSupValue : undefined;
            resourceInputs["csvNullValue"] = args ? args.csvNullValue : undefined;
            resourceInputs["csvRowDelimiter"] = args ? args.csvRowDelimiter : undefined;
            resourceInputs["dataFormat"] = args ? args.dataFormat : undefined;
            resourceInputs["dataPageSize"] = args ? args.dataPageSize : undefined;
            resourceInputs["datePartitionDelimiter"] = args ? args.datePartitionDelimiter : undefined;
            resourceInputs["datePartitionEnabled"] = args ? args.datePartitionEnabled : undefined;
            resourceInputs["datePartitionSequence"] = args ? args.datePartitionSequence : undefined;
            resourceInputs["datePartitionTimezone"] = args ? args.datePartitionTimezone : undefined;
            resourceInputs["detachTargetOnLobLookupFailureParquet"] = args ? args.detachTargetOnLobLookupFailureParquet : undefined;
            resourceInputs["dictPageSizeLimit"] = args ? args.dictPageSizeLimit : undefined;
            resourceInputs["enableStatistics"] = args ? args.enableStatistics : undefined;
            resourceInputs["encodingType"] = args ? args.encodingType : undefined;
            resourceInputs["encryptionMode"] = args ? args.encryptionMode : undefined;
            resourceInputs["endpointId"] = args ? args.endpointId : undefined;
            resourceInputs["endpointType"] = args ? args.endpointType : undefined;
            resourceInputs["expectedBucketOwner"] = args ? args.expectedBucketOwner : undefined;
            resourceInputs["externalTableDefinition"] = args ? args.externalTableDefinition : undefined;
            resourceInputs["ignoreHeaderRows"] = args ? args.ignoreHeaderRows : undefined;
            resourceInputs["includeOpForFullLoad"] = args ? args.includeOpForFullLoad : undefined;
            resourceInputs["kmsKeyArn"] = args ? args.kmsKeyArn : undefined;
            resourceInputs["maxFileSize"] = args ? args.maxFileSize : undefined;
            resourceInputs["parquetTimestampInMillisecond"] = args ? args.parquetTimestampInMillisecond : undefined;
            resourceInputs["parquetVersion"] = args ? args.parquetVersion : undefined;
            resourceInputs["preserveTransactions"] = args ? args.preserveTransactions : undefined;
            resourceInputs["rfc4180"] = args ? args.rfc4180 : undefined;
            resourceInputs["rowGroupLength"] = args ? args.rowGroupLength : undefined;
            resourceInputs["serverSideEncryptionKmsKeyId"] = args ? args.serverSideEncryptionKmsKeyId : undefined;
            resourceInputs["serviceAccessRoleArn"] = args ? args.serviceAccessRoleArn : undefined;
            resourceInputs["sslMode"] = args ? args.sslMode : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["timestampColumnName"] = args ? args.timestampColumnName : undefined;
            resourceInputs["useCsvNoSupValue"] = args ? args.useCsvNoSupValue : undefined;
            resourceInputs["useTaskStartTimeForFullLoadTimestamp"] = args ? args.useTaskStartTimeForFullLoadTimestamp : undefined;
            resourceInputs["endpointArn"] = undefined /*out*/;
            resourceInputs["engineDisplayName"] = undefined /*out*/;
            resourceInputs["externalId"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(S3Endpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.S3Endpoint = S3Endpoint;
/** @internal */
S3Endpoint.__pulumiType = 'aws:dms/s3Endpoint:S3Endpoint';
