// @pulumi/aws/mq/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInstanceTypeOfferingsOutput = exports.getInstanceTypeOfferings = exports.getBrokerOutput = exports.getBroker = exports.Configuration = exports.Broker = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Broker = null;
utilities.lazyLoad(exports, ["Broker"], () => require("./broker"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.getBroker = null;
exports.getBrokerOutput = null;
utilities.lazyLoad(exports, ["getBroker", "getBrokerOutput"], () => require("./getBroker"));
exports.getInstanceTypeOfferings = null;
exports.getInstanceTypeOfferingsOutput = null;
utilities.lazyLoad(exports, ["getInstanceTypeOfferings", "getInstanceTypeOfferingsOutput"], () => require("./getInstanceTypeOfferings"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:mq/broker:Broker":
                return new exports.Broker(name, undefined, { urn });
            case "aws:mq/configuration:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "mq/broker", _module);
pulumi.runtime.registerResourceModule("aws", "mq/configuration", _module);
