// @pulumi/aws/imagebuilder/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfrastructureConfiguration = exports.ImageRecipe = exports.ImagePipeline = exports.Image = exports.getInfrastructureConfigurationsOutput = exports.getInfrastructureConfigurations = exports.getInfrastructureConfigurationOutput = exports.getInfrastructureConfiguration = exports.getImageRecipesOutput = exports.getImageRecipes = exports.getImageRecipeOutput = exports.getImageRecipe = exports.getImagePipelinesOutput = exports.getImagePipelines = exports.getImagePipelineOutput = exports.getImagePipeline = exports.getImageOutput = exports.getImage = exports.getDistributionConfigurationsOutput = exports.getDistributionConfigurations = exports.getDistributionConfigurationOutput = exports.getDistributionConfiguration = exports.getContainerRecipesOutput = exports.getContainerRecipes = exports.getContainerRecipeOutput = exports.getContainerRecipe = exports.getComponentsOutput = exports.getComponents = exports.getComponentOutput = exports.getComponent = exports.DistributionConfiguration = exports.ContainerRecipe = exports.Component = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Component = null;
utilities.lazyLoad(exports, ["Component"], () => require("./component"));
exports.ContainerRecipe = null;
utilities.lazyLoad(exports, ["ContainerRecipe"], () => require("./containerRecipe"));
exports.DistributionConfiguration = null;
utilities.lazyLoad(exports, ["DistributionConfiguration"], () => require("./distributionConfiguration"));
exports.getComponent = null;
exports.getComponentOutput = null;
utilities.lazyLoad(exports, ["getComponent", "getComponentOutput"], () => require("./getComponent"));
exports.getComponents = null;
exports.getComponentsOutput = null;
utilities.lazyLoad(exports, ["getComponents", "getComponentsOutput"], () => require("./getComponents"));
exports.getContainerRecipe = null;
exports.getContainerRecipeOutput = null;
utilities.lazyLoad(exports, ["getContainerRecipe", "getContainerRecipeOutput"], () => require("./getContainerRecipe"));
exports.getContainerRecipes = null;
exports.getContainerRecipesOutput = null;
utilities.lazyLoad(exports, ["getContainerRecipes", "getContainerRecipesOutput"], () => require("./getContainerRecipes"));
exports.getDistributionConfiguration = null;
exports.getDistributionConfigurationOutput = null;
utilities.lazyLoad(exports, ["getDistributionConfiguration", "getDistributionConfigurationOutput"], () => require("./getDistributionConfiguration"));
exports.getDistributionConfigurations = null;
exports.getDistributionConfigurationsOutput = null;
utilities.lazyLoad(exports, ["getDistributionConfigurations", "getDistributionConfigurationsOutput"], () => require("./getDistributionConfigurations"));
exports.getImage = null;
exports.getImageOutput = null;
utilities.lazyLoad(exports, ["getImage", "getImageOutput"], () => require("./getImage"));
exports.getImagePipeline = null;
exports.getImagePipelineOutput = null;
utilities.lazyLoad(exports, ["getImagePipeline", "getImagePipelineOutput"], () => require("./getImagePipeline"));
exports.getImagePipelines = null;
exports.getImagePipelinesOutput = null;
utilities.lazyLoad(exports, ["getImagePipelines", "getImagePipelinesOutput"], () => require("./getImagePipelines"));
exports.getImageRecipe = null;
exports.getImageRecipeOutput = null;
utilities.lazyLoad(exports, ["getImageRecipe", "getImageRecipeOutput"], () => require("./getImageRecipe"));
exports.getImageRecipes = null;
exports.getImageRecipesOutput = null;
utilities.lazyLoad(exports, ["getImageRecipes", "getImageRecipesOutput"], () => require("./getImageRecipes"));
exports.getInfrastructureConfiguration = null;
exports.getInfrastructureConfigurationOutput = null;
utilities.lazyLoad(exports, ["getInfrastructureConfiguration", "getInfrastructureConfigurationOutput"], () => require("./getInfrastructureConfiguration"));
exports.getInfrastructureConfigurations = null;
exports.getInfrastructureConfigurationsOutput = null;
utilities.lazyLoad(exports, ["getInfrastructureConfigurations", "getInfrastructureConfigurationsOutput"], () => require("./getInfrastructureConfigurations"));
exports.Image = null;
utilities.lazyLoad(exports, ["Image"], () => require("./image"));
exports.ImagePipeline = null;
utilities.lazyLoad(exports, ["ImagePipeline"], () => require("./imagePipeline"));
exports.ImageRecipe = null;
utilities.lazyLoad(exports, ["ImageRecipe"], () => require("./imageRecipe"));
exports.InfrastructureConfiguration = null;
utilities.lazyLoad(exports, ["InfrastructureConfiguration"], () => require("./infrastructureConfiguration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:imagebuilder/component:Component":
                return new exports.Component(name, undefined, { urn });
            case "aws:imagebuilder/containerRecipe:ContainerRecipe":
                return new exports.ContainerRecipe(name, undefined, { urn });
            case "aws:imagebuilder/distributionConfiguration:DistributionConfiguration":
                return new exports.DistributionConfiguration(name, undefined, { urn });
            case "aws:imagebuilder/image:Image":
                return new exports.Image(name, undefined, { urn });
            case "aws:imagebuilder/imagePipeline:ImagePipeline":
                return new exports.ImagePipeline(name, undefined, { urn });
            case "aws:imagebuilder/imageRecipe:ImageRecipe":
                return new exports.ImageRecipe(name, undefined, { urn });
            case "aws:imagebuilder/infrastructureConfiguration:InfrastructureConfiguration":
                return new exports.InfrastructureConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "imagebuilder/component", _module);
pulumi.runtime.registerResourceModule("aws", "imagebuilder/containerRecipe", _module);
pulumi.runtime.registerResourceModule("aws", "imagebuilder/distributionConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "imagebuilder/image", _module);
pulumi.runtime.registerResourceModule("aws", "imagebuilder/imagePipeline", _module);
pulumi.runtime.registerResourceModule("aws", "imagebuilder/imageRecipe", _module);
pulumi.runtime.registerResourceModule("aws", "imagebuilder/infrastructureConfiguration", _module);
