// @pulumi/aws/datapipeline/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PipelineDefinition = exports.Pipeline = exports.getPipelineDefinitionOutput = exports.getPipelineDefinition = exports.getPipelineOutput = exports.getPipeline = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getPipeline = null;
exports.getPipelineOutput = null;
utilities.lazyLoad(exports, ["getPipeline", "getPipelineOutput"], () => require("./getPipeline"));
exports.getPipelineDefinition = null;
exports.getPipelineDefinitionOutput = null;
utilities.lazyLoad(exports, ["getPipelineDefinition", "getPipelineDefinitionOutput"], () => require("./getPipelineDefinition"));
exports.Pipeline = null;
utilities.lazyLoad(exports, ["Pipeline"], () => require("./pipeline"));
exports.PipelineDefinition = null;
utilities.lazyLoad(exports, ["PipelineDefinition"], () => require("./pipelineDefinition"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:datapipeline/pipeline:Pipeline":
                return new exports.Pipeline(name, undefined, { urn });
            case "aws:datapipeline/pipelineDefinition:PipelineDefinition":
                return new exports.PipelineDefinition(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "datapipeline/pipeline", _module);
pulumi.runtime.registerResourceModule("aws", "datapipeline/pipelineDefinition", _module);
