// @pulumi/aws/ses/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Template = exports.ReceiptRuleSet = exports.ReceiptRule = exports.ReceiptFilter = exports.MailFrom = exports.IdentityPolicy = exports.IdentityNotificationTopic = exports.getEmailIdentityOutput = exports.getEmailIdentity = exports.getDomainIdentityOutput = exports.getDomainIdentity = exports.getActiveReceiptRuleSet = exports.EventDestination = exports.EmailIdentity = exports.DomainIdentityVerification = exports.DomainIdentity = exports.DomainDkim = exports.ConfigurationSet = exports.ConfgurationSet = exports.ActiveReceiptRuleSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ActiveReceiptRuleSet = null;
utilities.lazyLoad(exports, ["ActiveReceiptRuleSet"], () => require("./activeReceiptRuleSet"));
exports.ConfgurationSet = null;
utilities.lazyLoad(exports, ["ConfgurationSet"], () => require("./confgurationSet"));
exports.ConfigurationSet = null;
utilities.lazyLoad(exports, ["ConfigurationSet"], () => require("./configurationSet"));
exports.DomainDkim = null;
utilities.lazyLoad(exports, ["DomainDkim"], () => require("./domainDkim"));
exports.DomainIdentity = null;
utilities.lazyLoad(exports, ["DomainIdentity"], () => require("./domainIdentity"));
exports.DomainIdentityVerification = null;
utilities.lazyLoad(exports, ["DomainIdentityVerification"], () => require("./domainIdentityVerification"));
exports.EmailIdentity = null;
utilities.lazyLoad(exports, ["EmailIdentity"], () => require("./emailIdentity"));
exports.EventDestination = null;
utilities.lazyLoad(exports, ["EventDestination"], () => require("./eventDestination"));
exports.getActiveReceiptRuleSet = null;
utilities.lazyLoad(exports, ["getActiveReceiptRuleSet"], () => require("./getActiveReceiptRuleSet"));
exports.getDomainIdentity = null;
exports.getDomainIdentityOutput = null;
utilities.lazyLoad(exports, ["getDomainIdentity", "getDomainIdentityOutput"], () => require("./getDomainIdentity"));
exports.getEmailIdentity = null;
exports.getEmailIdentityOutput = null;
utilities.lazyLoad(exports, ["getEmailIdentity", "getEmailIdentityOutput"], () => require("./getEmailIdentity"));
exports.IdentityNotificationTopic = null;
utilities.lazyLoad(exports, ["IdentityNotificationTopic"], () => require("./identityNotificationTopic"));
exports.IdentityPolicy = null;
utilities.lazyLoad(exports, ["IdentityPolicy"], () => require("./identityPolicy"));
exports.MailFrom = null;
utilities.lazyLoad(exports, ["MailFrom"], () => require("./mailFrom"));
exports.ReceiptFilter = null;
utilities.lazyLoad(exports, ["ReceiptFilter"], () => require("./receiptFilter"));
exports.ReceiptRule = null;
utilities.lazyLoad(exports, ["ReceiptRule"], () => require("./receiptRule"));
exports.ReceiptRuleSet = null;
utilities.lazyLoad(exports, ["ReceiptRuleSet"], () => require("./receiptRuleSet"));
exports.Template = null;
utilities.lazyLoad(exports, ["Template"], () => require("./template"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ses/activeReceiptRuleSet:ActiveReceiptRuleSet":
                return new exports.ActiveReceiptRuleSet(name, undefined, { urn });
            case "aws:ses/confgurationSet:ConfgurationSet":
                return new exports.ConfgurationSet(name, undefined, { urn });
            case "aws:ses/configurationSet:ConfigurationSet":
                return new exports.ConfigurationSet(name, undefined, { urn });
            case "aws:ses/domainDkim:DomainDkim":
                return new exports.DomainDkim(name, undefined, { urn });
            case "aws:ses/domainIdentity:DomainIdentity":
                return new exports.DomainIdentity(name, undefined, { urn });
            case "aws:ses/domainIdentityVerification:DomainIdentityVerification":
                return new exports.DomainIdentityVerification(name, undefined, { urn });
            case "aws:ses/emailIdentity:EmailIdentity":
                return new exports.EmailIdentity(name, undefined, { urn });
            case "aws:ses/eventDestination:EventDestination":
                return new exports.EventDestination(name, undefined, { urn });
            case "aws:ses/identityNotificationTopic:IdentityNotificationTopic":
                return new exports.IdentityNotificationTopic(name, undefined, { urn });
            case "aws:ses/identityPolicy:IdentityPolicy":
                return new exports.IdentityPolicy(name, undefined, { urn });
            case "aws:ses/mailFrom:MailFrom":
                return new exports.MailFrom(name, undefined, { urn });
            case "aws:ses/receiptFilter:ReceiptFilter":
                return new exports.ReceiptFilter(name, undefined, { urn });
            case "aws:ses/receiptRule:ReceiptRule":
                return new exports.ReceiptRule(name, undefined, { urn });
            case "aws:ses/receiptRuleSet:ReceiptRuleSet":
                return new exports.ReceiptRuleSet(name, undefined, { urn });
            case "aws:ses/template:Template":
                return new exports.Template(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ses/activeReceiptRuleSet", _module);
pulumi.runtime.registerResourceModule("aws", "ses/confgurationSet", _module);
pulumi.runtime.registerResourceModule("aws", "ses/configurationSet", _module);
pulumi.runtime.registerResourceModule("aws", "ses/domainDkim", _module);
pulumi.runtime.registerResourceModule("aws", "ses/domainIdentity", _module);
pulumi.runtime.registerResourceModule("aws", "ses/domainIdentityVerification", _module);
pulumi.runtime.registerResourceModule("aws", "ses/emailIdentity", _module);
pulumi.runtime.registerResourceModule("aws", "ses/eventDestination", _module);
pulumi.runtime.registerResourceModule("aws", "ses/identityNotificationTopic", _module);
pulumi.runtime.registerResourceModule("aws", "ses/identityPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "ses/mailFrom", _module);
pulumi.runtime.registerResourceModule("aws", "ses/receiptFilter", _module);
pulumi.runtime.registerResourceModule("aws", "ses/receiptRule", _module);
pulumi.runtime.registerResourceModule("aws", "ses/receiptRuleSet", _module);
pulumi.runtime.registerResourceModule("aws", "ses/template", _module);
