// @pulumi/aws/signer/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SigningProfilePermission = exports.SigningProfile = exports.SigningJob = exports.getSigningProfileOutput = exports.getSigningProfile = exports.getSigningJobOutput = exports.getSigningJob = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getSigningJob = null;
exports.getSigningJobOutput = null;
utilities.lazyLoad(exports, ["getSigningJob", "getSigningJobOutput"], () => require("./getSigningJob"));
exports.getSigningProfile = null;
exports.getSigningProfileOutput = null;
utilities.lazyLoad(exports, ["getSigningProfile", "getSigningProfileOutput"], () => require("./getSigningProfile"));
exports.SigningJob = null;
utilities.lazyLoad(exports, ["SigningJob"], () => require("./signingJob"));
exports.SigningProfile = null;
utilities.lazyLoad(exports, ["SigningProfile"], () => require("./signingProfile"));
exports.SigningProfilePermission = null;
utilities.lazyLoad(exports, ["SigningProfilePermission"], () => require("./signingProfilePermission"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:signer/signingJob:SigningJob":
                return new exports.SigningJob(name, undefined, { urn });
            case "aws:signer/signingProfile:SigningProfile":
                return new exports.SigningProfile(name, undefined, { urn });
            case "aws:signer/signingProfilePermission:SigningProfilePermission":
                return new exports.SigningProfilePermission(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "signer/signingJob", _module);
pulumi.runtime.registerResourceModule("aws", "signer/signingProfile", _module);
pulumi.runtime.registerResourceModule("aws", "signer/signingProfilePermission", _module);
