// @pulumi/aws/oam/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SinkPolicy = exports.Sink = exports.Link = exports.getSinks = exports.getSinkOutput = exports.getSink = exports.getLinks = exports.getLinkOutput = exports.getLink = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getLink = null;
exports.getLinkOutput = null;
utilities.lazyLoad(exports, ["getLink", "getLinkOutput"], () => require("./getLink"));
exports.getLinks = null;
utilities.lazyLoad(exports, ["getLinks"], () => require("./getLinks"));
exports.getSink = null;
exports.getSinkOutput = null;
utilities.lazyLoad(exports, ["getSink", "getSinkOutput"], () => require("./getSink"));
exports.getSinks = null;
utilities.lazyLoad(exports, ["getSinks"], () => require("./getSinks"));
exports.Link = null;
utilities.lazyLoad(exports, ["Link"], () => require("./link"));
exports.Sink = null;
utilities.lazyLoad(exports, ["Sink"], () => require("./sink"));
exports.SinkPolicy = null;
utilities.lazyLoad(exports, ["SinkPolicy"], () => require("./sinkPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:oam/link:Link":
                return new exports.Link(name, undefined, { urn });
            case "aws:oam/sink:Sink":
                return new exports.Sink(name, undefined, { urn });
            case "aws:oam/sinkPolicy:SinkPolicy":
                return new exports.SinkPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "oam/link", _module);
pulumi.runtime.registerResourceModule("aws", "oam/sink", _module);
pulumi.runtime.registerResourceModule("aws", "oam/sinkPolicy", _module);
