// @pulumi/aws/pinpoint/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmsChannel = exports.GcmChannel = exports.EventStream = exports.EmailChannel = exports.BaiduChannel = exports.App = exports.ApnsVoipSandboxChannel = exports.ApnsVoipChannel = exports.ApnsSandboxChannel = exports.ApnsChannel = exports.AdmChannel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AdmChannel = null;
utilities.lazyLoad(exports, ["AdmChannel"], () => require("./admChannel"));
exports.ApnsChannel = null;
utilities.lazyLoad(exports, ["ApnsChannel"], () => require("./apnsChannel"));
exports.ApnsSandboxChannel = null;
utilities.lazyLoad(exports, ["ApnsSandboxChannel"], () => require("./apnsSandboxChannel"));
exports.ApnsVoipChannel = null;
utilities.lazyLoad(exports, ["ApnsVoipChannel"], () => require("./apnsVoipChannel"));
exports.ApnsVoipSandboxChannel = null;
utilities.lazyLoad(exports, ["ApnsVoipSandboxChannel"], () => require("./apnsVoipSandboxChannel"));
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.BaiduChannel = null;
utilities.lazyLoad(exports, ["BaiduChannel"], () => require("./baiduChannel"));
exports.EmailChannel = null;
utilities.lazyLoad(exports, ["EmailChannel"], () => require("./emailChannel"));
exports.EventStream = null;
utilities.lazyLoad(exports, ["EventStream"], () => require("./eventStream"));
exports.GcmChannel = null;
utilities.lazyLoad(exports, ["GcmChannel"], () => require("./gcmChannel"));
exports.SmsChannel = null;
utilities.lazyLoad(exports, ["SmsChannel"], () => require("./smsChannel"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:pinpoint/admChannel:AdmChannel":
                return new exports.AdmChannel(name, undefined, { urn });
            case "aws:pinpoint/apnsChannel:ApnsChannel":
                return new exports.ApnsChannel(name, undefined, { urn });
            case "aws:pinpoint/apnsSandboxChannel:ApnsSandboxChannel":
                return new exports.ApnsSandboxChannel(name, undefined, { urn });
            case "aws:pinpoint/apnsVoipChannel:ApnsVoipChannel":
                return new exports.ApnsVoipChannel(name, undefined, { urn });
            case "aws:pinpoint/apnsVoipSandboxChannel:ApnsVoipSandboxChannel":
                return new exports.ApnsVoipSandboxChannel(name, undefined, { urn });
            case "aws:pinpoint/app:App":
                return new exports.App(name, undefined, { urn });
            case "aws:pinpoint/baiduChannel:BaiduChannel":
                return new exports.BaiduChannel(name, undefined, { urn });
            case "aws:pinpoint/emailChannel:EmailChannel":
                return new exports.EmailChannel(name, undefined, { urn });
            case "aws:pinpoint/eventStream:EventStream":
                return new exports.EventStream(name, undefined, { urn });
            case "aws:pinpoint/gcmChannel:GcmChannel":
                return new exports.GcmChannel(name, undefined, { urn });
            case "aws:pinpoint/smsChannel:SmsChannel":
                return new exports.SmsChannel(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "pinpoint/admChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/apnsChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/apnsSandboxChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/apnsVoipChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/apnsVoipSandboxChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/app", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/baiduChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/emailChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/eventStream", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/gcmChannel", _module);
pulumi.runtime.registerResourceModule("aws", "pinpoint/smsChannel", _module);
