// @pulumi/aws/inspector2/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationConfiguration = exports.MemberAssociation = exports.Enabler = exports.DelegatedAdminAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DelegatedAdminAccount = null;
utilities.lazyLoad(exports, ["DelegatedAdminAccount"], () => require("./delegatedAdminAccount"));
exports.Enabler = null;
utilities.lazyLoad(exports, ["Enabler"], () => require("./enabler"));
exports.MemberAssociation = null;
utilities.lazyLoad(exports, ["MemberAssociation"], () => require("./memberAssociation"));
exports.OrganizationConfiguration = null;
utilities.lazyLoad(exports, ["OrganizationConfiguration"], () => require("./organizationConfiguration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:inspector2/delegatedAdminAccount:DelegatedAdminAccount":
                return new exports.DelegatedAdminAccount(name, undefined, { urn });
            case "aws:inspector2/enabler:Enabler":
                return new exports.Enabler(name, undefined, { urn });
            case "aws:inspector2/memberAssociation:MemberAssociation":
                return new exports.MemberAssociation(name, undefined, { urn });
            case "aws:inspector2/organizationConfiguration:OrganizationConfiguration":
                return new exports.OrganizationConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "inspector2/delegatedAdminAccount", _module);
pulumi.runtime.registerResourceModule("aws", "inspector2/enabler", _module);
pulumi.runtime.registerResourceModule("aws", "inspector2/memberAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "inspector2/organizationConfiguration", _module);
