// @pulumi/aws/licensemanager/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseGrantAccepter = exports.LicenseGrant = exports.LicenseConfiguration = exports.getReceivedLicensesOutput = exports.getReceivedLicenses = exports.getReceivedLicenseOutput = exports.getReceivedLicense = exports.getLicenseGrantsOutput = exports.getLicenseGrants = exports.Association = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Association = null;
utilities.lazyLoad(exports, ["Association"], () => require("./association"));
exports.getLicenseGrants = null;
exports.getLicenseGrantsOutput = null;
utilities.lazyLoad(exports, ["getLicenseGrants", "getLicenseGrantsOutput"], () => require("./getLicenseGrants"));
exports.getReceivedLicense = null;
exports.getReceivedLicenseOutput = null;
utilities.lazyLoad(exports, ["getReceivedLicense", "getReceivedLicenseOutput"], () => require("./getReceivedLicense"));
exports.getReceivedLicenses = null;
exports.getReceivedLicensesOutput = null;
utilities.lazyLoad(exports, ["getReceivedLicenses", "getReceivedLicensesOutput"], () => require("./getReceivedLicenses"));
exports.LicenseConfiguration = null;
utilities.lazyLoad(exports, ["LicenseConfiguration"], () => require("./licenseConfiguration"));
exports.LicenseGrant = null;
utilities.lazyLoad(exports, ["LicenseGrant"], () => require("./licenseGrant"));
exports.LicenseGrantAccepter = null;
utilities.lazyLoad(exports, ["LicenseGrantAccepter"], () => require("./licenseGrantAccepter"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:licensemanager/association:Association":
                return new exports.Association(name, undefined, { urn });
            case "aws:licensemanager/licenseConfiguration:LicenseConfiguration":
                return new exports.LicenseConfiguration(name, undefined, { urn });
            case "aws:licensemanager/licenseGrant:LicenseGrant":
                return new exports.LicenseGrant(name, undefined, { urn });
            case "aws:licensemanager/licenseGrantAccepter:LicenseGrantAccepter":
                return new exports.LicenseGrantAccepter(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "licensemanager/association", _module);
pulumi.runtime.registerResourceModule("aws", "licensemanager/licenseConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "licensemanager/licenseGrant", _module);
pulumi.runtime.registerResourceModule("aws", "licensemanager/licenseGrantAccepter", _module);
