// @pulumi/aws/mskconnect/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkerConfiguration = exports.getWorkerConfigurationOutput = exports.getWorkerConfiguration = exports.getCustomPluginOutput = exports.getCustomPlugin = exports.getConnectorOutput = exports.getConnector = exports.CustomPlugin = exports.Connector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connector = null;
utilities.lazyLoad(exports, ["Connector"], () => require("./connector"));
exports.CustomPlugin = null;
utilities.lazyLoad(exports, ["CustomPlugin"], () => require("./customPlugin"));
exports.getConnector = null;
exports.getConnectorOutput = null;
utilities.lazyLoad(exports, ["getConnector", "getConnectorOutput"], () => require("./getConnector"));
exports.getCustomPlugin = null;
exports.getCustomPluginOutput = null;
utilities.lazyLoad(exports, ["getCustomPlugin", "getCustomPluginOutput"], () => require("./getCustomPlugin"));
exports.getWorkerConfiguration = null;
exports.getWorkerConfigurationOutput = null;
utilities.lazyLoad(exports, ["getWorkerConfiguration", "getWorkerConfigurationOutput"], () => require("./getWorkerConfiguration"));
exports.WorkerConfiguration = null;
utilities.lazyLoad(exports, ["WorkerConfiguration"], () => require("./workerConfiguration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:mskconnect/connector:Connector":
                return new exports.Connector(name, undefined, { urn });
            case "aws:mskconnect/customPlugin:CustomPlugin":
                return new exports.CustomPlugin(name, undefined, { urn });
            case "aws:mskconnect/workerConfiguration:WorkerConfiguration":
                return new exports.WorkerConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "mskconnect/connector", _module);
pulumi.runtime.registerResourceModule("aws", "mskconnect/customPlugin", _module);
pulumi.runtime.registerResourceModule("aws", "mskconnect/workerConfiguration", _module);
