// @pulumi/aws/codestarconnections/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Host = exports.getConnectionOutput = exports.getConnection = exports.Connection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.getConnection = null;
exports.getConnectionOutput = null;
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
exports.Host = null;
utilities.lazyLoad(exports, ["Host"], () => require("./host"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:codestarconnections/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "aws:codestarconnections/host:Host":
                return new exports.Host(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "codestarconnections/connection", _module);
pulumi.runtime.registerResourceModule("aws", "codestarconnections/host", _module);
