// @pulumi/aws/synthetics/groupAssociation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a Synthetics Group Association resource.
 *
 * ## Example Usage
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.synthetics.GroupAssociation("example", {
 *     groupName: aws_synthetics_group.example.name,
 *     canaryArn: aws_synthetics_canary.example.arn,
 * });
 * ```
 *
 * ## Import
 *
 * CloudWatch Synthetics Group Association can be imported in the form `canary_arn,group_name`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:synthetics/groupAssociation:GroupAssociation example arn:aws:synthetics:us-west-2:123456789012:canary:tf-acc-test-abcd1234,examplename
 * ```
 */
class GroupAssociation extends pulumi.CustomResource {
    /**
     * Get an existing GroupAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new GroupAssociation(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GroupAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GroupAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["canaryArn"] = state ? state.canaryArn : undefined;
            resourceInputs["groupArn"] = state ? state.groupArn : undefined;
            resourceInputs["groupId"] = state ? state.groupId : undefined;
            resourceInputs["groupName"] = state ? state.groupName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.canaryArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'canaryArn'");
            }
            if ((!args || args.groupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupName'");
            }
            resourceInputs["canaryArn"] = args ? args.canaryArn : undefined;
            resourceInputs["groupName"] = args ? args.groupName : undefined;
            resourceInputs["groupArn"] = undefined /*out*/;
            resourceInputs["groupId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(GroupAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GroupAssociation = GroupAssociation;
/** @internal */
GroupAssociation.__pulumiType = 'aws:synthetics/groupAssociation:GroupAssociation';
