// @pulumi/aws/ec2clientvpn/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Route = exports.NetworkAssociation = exports.getEndpointOutput = exports.getEndpoint = exports.Endpoint = exports.AuthorizationRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AuthorizationRule = null;
utilities.lazyLoad(exports, ["AuthorizationRule"], () => require("./authorizationRule"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.NetworkAssociation = null;
utilities.lazyLoad(exports, ["NetworkAssociation"], () => require("./networkAssociation"));
exports.Route = null;
utilities.lazyLoad(exports, ["Route"], () => require("./route"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ec2clientvpn/authorizationRule:AuthorizationRule":
                return new exports.AuthorizationRule(name, undefined, { urn });
            case "aws:ec2clientvpn/endpoint:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "aws:ec2clientvpn/networkAssociation:NetworkAssociation":
                return new exports.NetworkAssociation(name, undefined, { urn });
            case "aws:ec2clientvpn/route:Route":
                return new exports.Route(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/authorizationRule", _module);
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/endpoint", _module);
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/networkAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/route", _module);
