// @pulumi/aws/cloudfront/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseHeadersPolicy = exports.RealtimeLogConfig = exports.PublicKey = exports.OriginRequestPolicy = exports.OriginAccessIdentity = exports.OriginAccessControl = exports.MonitoringSubscription = exports.KeyGroup = exports.getResponseHeadersPolicyOutput = exports.getResponseHeadersPolicy = exports.getRealtimeLogConfigOutput = exports.getRealtimeLogConfig = exports.getOriginRequestPolicyOutput = exports.getOriginRequestPolicy = exports.getOriginAccessIdentityOutput = exports.getOriginAccessIdentity = exports.getOriginAccessIdentitiesOutput = exports.getOriginAccessIdentities = exports.getLogDeliveryCanonicalUserIdOutput = exports.getLogDeliveryCanonicalUserId = exports.getFunctionOutput = exports.getFunction = exports.getDistributionOutput = exports.getDistribution = exports.getCachePolicyOutput = exports.getCachePolicy = exports.Function = exports.FieldLevelEncryptionProfile = exports.FieldLevelEncryptionConfig = exports.Distribution = exports.CachePolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CachePolicy = null;
utilities.lazyLoad(exports, ["CachePolicy"], () => require("./cachePolicy"));
exports.Distribution = null;
utilities.lazyLoad(exports, ["Distribution"], () => require("./distribution"));
exports.FieldLevelEncryptionConfig = null;
utilities.lazyLoad(exports, ["FieldLevelEncryptionConfig"], () => require("./fieldLevelEncryptionConfig"));
exports.FieldLevelEncryptionProfile = null;
utilities.lazyLoad(exports, ["FieldLevelEncryptionProfile"], () => require("./fieldLevelEncryptionProfile"));
exports.Function = null;
utilities.lazyLoad(exports, ["Function"], () => require("./function"));
exports.getCachePolicy = null;
exports.getCachePolicyOutput = null;
utilities.lazyLoad(exports, ["getCachePolicy", "getCachePolicyOutput"], () => require("./getCachePolicy"));
exports.getDistribution = null;
exports.getDistributionOutput = null;
utilities.lazyLoad(exports, ["getDistribution", "getDistributionOutput"], () => require("./getDistribution"));
exports.getFunction = null;
exports.getFunctionOutput = null;
utilities.lazyLoad(exports, ["getFunction", "getFunctionOutput"], () => require("./getFunction"));
exports.getLogDeliveryCanonicalUserId = null;
exports.getLogDeliveryCanonicalUserIdOutput = null;
utilities.lazyLoad(exports, ["getLogDeliveryCanonicalUserId", "getLogDeliveryCanonicalUserIdOutput"], () => require("./getLogDeliveryCanonicalUserId"));
exports.getOriginAccessIdentities = null;
exports.getOriginAccessIdentitiesOutput = null;
utilities.lazyLoad(exports, ["getOriginAccessIdentities", "getOriginAccessIdentitiesOutput"], () => require("./getOriginAccessIdentities"));
exports.getOriginAccessIdentity = null;
exports.getOriginAccessIdentityOutput = null;
utilities.lazyLoad(exports, ["getOriginAccessIdentity", "getOriginAccessIdentityOutput"], () => require("./getOriginAccessIdentity"));
exports.getOriginRequestPolicy = null;
exports.getOriginRequestPolicyOutput = null;
utilities.lazyLoad(exports, ["getOriginRequestPolicy", "getOriginRequestPolicyOutput"], () => require("./getOriginRequestPolicy"));
exports.getRealtimeLogConfig = null;
exports.getRealtimeLogConfigOutput = null;
utilities.lazyLoad(exports, ["getRealtimeLogConfig", "getRealtimeLogConfigOutput"], () => require("./getRealtimeLogConfig"));
exports.getResponseHeadersPolicy = null;
exports.getResponseHeadersPolicyOutput = null;
utilities.lazyLoad(exports, ["getResponseHeadersPolicy", "getResponseHeadersPolicyOutput"], () => require("./getResponseHeadersPolicy"));
exports.KeyGroup = null;
utilities.lazyLoad(exports, ["KeyGroup"], () => require("./keyGroup"));
exports.MonitoringSubscription = null;
utilities.lazyLoad(exports, ["MonitoringSubscription"], () => require("./monitoringSubscription"));
exports.OriginAccessControl = null;
utilities.lazyLoad(exports, ["OriginAccessControl"], () => require("./originAccessControl"));
exports.OriginAccessIdentity = null;
utilities.lazyLoad(exports, ["OriginAccessIdentity"], () => require("./originAccessIdentity"));
exports.OriginRequestPolicy = null;
utilities.lazyLoad(exports, ["OriginRequestPolicy"], () => require("./originRequestPolicy"));
exports.PublicKey = null;
utilities.lazyLoad(exports, ["PublicKey"], () => require("./publicKey"));
exports.RealtimeLogConfig = null;
utilities.lazyLoad(exports, ["RealtimeLogConfig"], () => require("./realtimeLogConfig"));
exports.ResponseHeadersPolicy = null;
utilities.lazyLoad(exports, ["ResponseHeadersPolicy"], () => require("./responseHeadersPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:cloudfront/cachePolicy:CachePolicy":
                return new exports.CachePolicy(name, undefined, { urn });
            case "aws:cloudfront/distribution:Distribution":
                return new exports.Distribution(name, undefined, { urn });
            case "aws:cloudfront/fieldLevelEncryptionConfig:FieldLevelEncryptionConfig":
                return new exports.FieldLevelEncryptionConfig(name, undefined, { urn });
            case "aws:cloudfront/fieldLevelEncryptionProfile:FieldLevelEncryptionProfile":
                return new exports.FieldLevelEncryptionProfile(name, undefined, { urn });
            case "aws:cloudfront/function:Function":
                return new exports.Function(name, undefined, { urn });
            case "aws:cloudfront/keyGroup:KeyGroup":
                return new exports.KeyGroup(name, undefined, { urn });
            case "aws:cloudfront/monitoringSubscription:MonitoringSubscription":
                return new exports.MonitoringSubscription(name, undefined, { urn });
            case "aws:cloudfront/originAccessControl:OriginAccessControl":
                return new exports.OriginAccessControl(name, undefined, { urn });
            case "aws:cloudfront/originAccessIdentity:OriginAccessIdentity":
                return new exports.OriginAccessIdentity(name, undefined, { urn });
            case "aws:cloudfront/originRequestPolicy:OriginRequestPolicy":
                return new exports.OriginRequestPolicy(name, undefined, { urn });
            case "aws:cloudfront/publicKey:PublicKey":
                return new exports.PublicKey(name, undefined, { urn });
            case "aws:cloudfront/realtimeLogConfig:RealtimeLogConfig":
                return new exports.RealtimeLogConfig(name, undefined, { urn });
            case "aws:cloudfront/responseHeadersPolicy:ResponseHeadersPolicy":
                return new exports.ResponseHeadersPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "cloudfront/cachePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/distribution", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/fieldLevelEncryptionConfig", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/fieldLevelEncryptionProfile", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/function", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/keyGroup", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/monitoringSubscription", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/originAccessControl", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/originAccessIdentity", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/originRequestPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/publicKey", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/realtimeLogConfig", _module);
pulumi.runtime.registerResourceModule("aws", "cloudfront/responseHeadersPolicy", _module);
