// @pulumi/aws/chime/sdkvoiceVoiceProfileDomain.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SdkvoiceVoiceProfileDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS Chime SDK Voice Profile Domain.
 *
 * ## Example Usage
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleKey = new aws.kms.Key("exampleKey", {
 *     description: "KMS Key for Voice Profile Domain",
 *     deletionWindowInDays: 7,
 * });
 * const exampleSdkvoiceVoiceProfileDomain = new aws.chime.SdkvoiceVoiceProfileDomain("exampleSdkvoiceVoiceProfileDomain", {
 *     serverSideEncryptionConfiguration: {
 *         kmsKeyArn: exampleKey.arn,
 *     },
 *     description: "My Voice Profile Domain",
 *     tags: {
 *         key1: "value1",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * AWS Chime SDK Voice Profile Domain can be imported using the `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:chime/sdkvoiceVoiceProfileDomain:SdkvoiceVoiceProfileDomain example abcdef123456
 * ```
 */
class SdkvoiceVoiceProfileDomain extends pulumi.CustomResource {
    /**
     * Get an existing SdkvoiceVoiceProfileDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SdkvoiceVoiceProfileDomain(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SdkvoiceVoiceProfileDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SdkvoiceVoiceProfileDomain.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["serverSideEncryptionConfiguration"] = state ? state.serverSideEncryptionConfiguration : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.serverSideEncryptionConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverSideEncryptionConfiguration'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["serverSideEncryptionConfiguration"] = args ? args.serverSideEncryptionConfiguration : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SdkvoiceVoiceProfileDomain.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SdkvoiceVoiceProfileDomain = SdkvoiceVoiceProfileDomain;
/** @internal */
SdkvoiceVoiceProfileDomain.__pulumiType = 'aws:chime/sdkvoiceVoiceProfileDomain:SdkvoiceVoiceProfileDomain';
