// @pulumi/aws/evidently/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Segment = exports.Project = exports.Launch = exports.Feature = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Feature = null;
utilities.lazyLoad(exports, ["Feature"], () => require("./feature"));
exports.Launch = null;
utilities.lazyLoad(exports, ["Launch"], () => require("./launch"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.Segment = null;
utilities.lazyLoad(exports, ["Segment"], () => require("./segment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:evidently/feature:Feature":
                return new exports.Feature(name, undefined, { urn });
            case "aws:evidently/launch:Launch":
                return new exports.Launch(name, undefined, { urn });
            case "aws:evidently/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws:evidently/segment:Segment":
                return new exports.Segment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "evidently/feature", _module);
pulumi.runtime.registerResourceModule("aws", "evidently/launch", _module);
pulumi.runtime.registerResourceModule("aws", "evidently/project", _module);
pulumi.runtime.registerResourceModule("aws", "evidently/segment", _module);
