// @pulumi/aws/medialive/input.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Input = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS MediaLive Input.
 *
 * ## Example Usage
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleInputSecurityGroup = new aws.medialive.InputSecurityGroup("exampleInputSecurityGroup", {
 *     whitelistRules: [{
 *         cidr: "10.0.0.8/32",
 *     }],
 *     tags: {
 *         ENVIRONMENT: "prod",
 *     },
 * });
 * const exampleInput = new aws.medialive.Input("exampleInput", {
 *     inputSecurityGroups: [exampleInputSecurityGroup.id],
 *     type: "UDP_PUSH",
 *     tags: {
 *         ENVIRONMENT: "prod",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * MediaLive Input can be imported using the `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:medialive/input:Input example 12345678
 * ```
 */
class Input extends pulumi.CustomResource {
    /**
     * Get an existing Input resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Input(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Input.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Input.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["attachedChannels"] = state ? state.attachedChannels : undefined;
            resourceInputs["destinations"] = state ? state.destinations : undefined;
            resourceInputs["inputClass"] = state ? state.inputClass : undefined;
            resourceInputs["inputDevices"] = state ? state.inputDevices : undefined;
            resourceInputs["inputPartnerIds"] = state ? state.inputPartnerIds : undefined;
            resourceInputs["inputSecurityGroups"] = state ? state.inputSecurityGroups : undefined;
            resourceInputs["inputSourceType"] = state ? state.inputSourceType : undefined;
            resourceInputs["mediaConnectFlows"] = state ? state.mediaConnectFlows : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["roleArn"] = state ? state.roleArn : undefined;
            resourceInputs["sources"] = state ? state.sources : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
            resourceInputs["vpc"] = state ? state.vpc : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["destinations"] = args ? args.destinations : undefined;
            resourceInputs["inputDevices"] = args ? args.inputDevices : undefined;
            resourceInputs["inputSecurityGroups"] = args ? args.inputSecurityGroups : undefined;
            resourceInputs["mediaConnectFlows"] = args ? args.mediaConnectFlows : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["sources"] = args ? args.sources : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["vpc"] = args ? args.vpc : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["attachedChannels"] = undefined /*out*/;
            resourceInputs["inputClass"] = undefined /*out*/;
            resourceInputs["inputPartnerIds"] = undefined /*out*/;
            resourceInputs["inputSourceType"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Input.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Input = Input;
/** @internal */
Input.__pulumiType = 'aws:medialive/input:Input';
