// @pulumi/aws/ebs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Volume = exports.SnapshotImport = exports.SnapshotCopy = exports.Snapshot = exports.getVolumeOutput = exports.getVolume = exports.getSnapshotIdsOutput = exports.getSnapshotIds = exports.getSnapshotOutput = exports.getSnapshot = exports.getEncryptionByDefault = exports.getEbsVolumesOutput = exports.getEbsVolumes = exports.getDefaultKmsKey = exports.EncryptionByDefault = exports.DefaultKmsKey = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DefaultKmsKey = null;
utilities.lazyLoad(exports, ["DefaultKmsKey"], () => require("./defaultKmsKey"));
exports.EncryptionByDefault = null;
utilities.lazyLoad(exports, ["EncryptionByDefault"], () => require("./encryptionByDefault"));
exports.getDefaultKmsKey = null;
utilities.lazyLoad(exports, ["getDefaultKmsKey"], () => require("./getDefaultKmsKey"));
exports.getEbsVolumes = null;
exports.getEbsVolumesOutput = null;
utilities.lazyLoad(exports, ["getEbsVolumes", "getEbsVolumesOutput"], () => require("./getEbsVolumes"));
exports.getEncryptionByDefault = null;
utilities.lazyLoad(exports, ["getEncryptionByDefault"], () => require("./getEncryptionByDefault"));
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
exports.getSnapshotIds = null;
exports.getSnapshotIdsOutput = null;
utilities.lazyLoad(exports, ["getSnapshotIds", "getSnapshotIdsOutput"], () => require("./getSnapshotIds"));
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.SnapshotCopy = null;
utilities.lazyLoad(exports, ["SnapshotCopy"], () => require("./snapshotCopy"));
exports.SnapshotImport = null;
utilities.lazyLoad(exports, ["SnapshotImport"], () => require("./snapshotImport"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ebs/defaultKmsKey:DefaultKmsKey":
                return new exports.DefaultKmsKey(name, undefined, { urn });
            case "aws:ebs/encryptionByDefault:EncryptionByDefault":
                return new exports.EncryptionByDefault(name, undefined, { urn });
            case "aws:ebs/snapshot:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "aws:ebs/snapshotCopy:SnapshotCopy":
                return new exports.SnapshotCopy(name, undefined, { urn });
            case "aws:ebs/snapshotImport:SnapshotImport":
                return new exports.SnapshotImport(name, undefined, { urn });
            case "aws:ebs/volume:Volume":
                return new exports.Volume(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ebs/defaultKmsKey", _module);
pulumi.runtime.registerResourceModule("aws", "ebs/encryptionByDefault", _module);
pulumi.runtime.registerResourceModule("aws", "ebs/snapshot", _module);
pulumi.runtime.registerResourceModule("aws", "ebs/snapshotCopy", _module);
pulumi.runtime.registerResourceModule("aws", "ebs/snapshotImport", _module);
pulumi.runtime.registerResourceModule("aws", "ebs/volume", _module);
