// @pulumi/aws/devicefarm/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Upload = exports.TestGridProject = exports.Project = exports.NetworkProfile = exports.InstanceProfile = exports.DevicePool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DevicePool = null;
utilities.lazyLoad(exports, ["DevicePool"], () => require("./devicePool"));
exports.InstanceProfile = null;
utilities.lazyLoad(exports, ["InstanceProfile"], () => require("./instanceProfile"));
exports.NetworkProfile = null;
utilities.lazyLoad(exports, ["NetworkProfile"], () => require("./networkProfile"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.TestGridProject = null;
utilities.lazyLoad(exports, ["TestGridProject"], () => require("./testGridProject"));
exports.Upload = null;
utilities.lazyLoad(exports, ["Upload"], () => require("./upload"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:devicefarm/devicePool:DevicePool":
                return new exports.DevicePool(name, undefined, { urn });
            case "aws:devicefarm/instanceProfile:InstanceProfile":
                return new exports.InstanceProfile(name, undefined, { urn });
            case "aws:devicefarm/networkProfile:NetworkProfile":
                return new exports.NetworkProfile(name, undefined, { urn });
            case "aws:devicefarm/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws:devicefarm/testGridProject:TestGridProject":
                return new exports.TestGridProject(name, undefined, { urn });
            case "aws:devicefarm/upload:Upload":
                return new exports.Upload(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "devicefarm/devicePool", _module);
pulumi.runtime.registerResourceModule("aws", "devicefarm/instanceProfile", _module);
pulumi.runtime.registerResourceModule("aws", "devicefarm/networkProfile", _module);
pulumi.runtime.registerResourceModule("aws", "devicefarm/project", _module);
pulumi.runtime.registerResourceModule("aws", "devicefarm/testGridProject", _module);
pulumi.runtime.registerResourceModule("aws", "devicefarm/upload", _module);
