// @pulumi/aws/ram/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceShareAccepter = exports.ResourceShare = exports.ResourceAssociation = exports.PrincipalAssociation = exports.getResourceShareOutput = exports.getResourceShare = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getResourceShare = null;
exports.getResourceShareOutput = null;
utilities.lazyLoad(exports, ["getResourceShare", "getResourceShareOutput"], () => require("./getResourceShare"));
exports.PrincipalAssociation = null;
utilities.lazyLoad(exports, ["PrincipalAssociation"], () => require("./principalAssociation"));
exports.ResourceAssociation = null;
utilities.lazyLoad(exports, ["ResourceAssociation"], () => require("./resourceAssociation"));
exports.ResourceShare = null;
utilities.lazyLoad(exports, ["ResourceShare"], () => require("./resourceShare"));
exports.ResourceShareAccepter = null;
utilities.lazyLoad(exports, ["ResourceShareAccepter"], () => require("./resourceShareAccepter"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ram/principalAssociation:PrincipalAssociation":
                return new exports.PrincipalAssociation(name, undefined, { urn });
            case "aws:ram/resourceAssociation:ResourceAssociation":
                return new exports.ResourceAssociation(name, undefined, { urn });
            case "aws:ram/resourceShare:ResourceShare":
                return new exports.ResourceShare(name, undefined, { urn });
            case "aws:ram/resourceShareAccepter:ResourceShareAccepter":
                return new exports.ResourceShareAccepter(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ram/principalAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "ram/resourceAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "ram/resourceShare", _module);
pulumi.runtime.registerResourceModule("aws", "ram/resourceShareAccepter", _module);
