// @pulumi/aws/ssmincidents/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponsePlan = exports.ReplicationSet = exports.getResponsePlanOutput = exports.getResponsePlan = exports.getReplicationSetOutput = exports.getReplicationSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getReplicationSet = null;
exports.getReplicationSetOutput = null;
utilities.lazyLoad(exports, ["getReplicationSet", "getReplicationSetOutput"], () => require("./getReplicationSet"));
exports.getResponsePlan = null;
exports.getResponsePlanOutput = null;
utilities.lazyLoad(exports, ["getResponsePlan", "getResponsePlanOutput"], () => require("./getResponsePlan"));
exports.ReplicationSet = null;
utilities.lazyLoad(exports, ["ReplicationSet"], () => require("./replicationSet"));
exports.ResponsePlan = null;
utilities.lazyLoad(exports, ["ResponsePlan"], () => require("./responsePlan"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ssmincidents/replicationSet:ReplicationSet":
                return new exports.ReplicationSet(name, undefined, { urn });
            case "aws:ssmincidents/responsePlan:ResponsePlan":
                return new exports.ResponsePlan(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ssmincidents/replicationSet", _module);
pulumi.runtime.registerResourceModule("aws", "ssmincidents/responsePlan", _module);
