// @pulumi/aws/dms/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3Endpoint = exports.ReplicationTask = exports.ReplicationSubnetGroup = exports.ReplicationInstance = exports.getReplicationTaskOutput = exports.getReplicationTask = exports.getReplicationSubnetGroupOutput = exports.getReplicationSubnetGroup = exports.getReplicationInstanceOutput = exports.getReplicationInstance = exports.getEndpointOutput = exports.getEndpoint = exports.getCertificateOutput = exports.getCertificate = exports.EventSubscription = exports.Endpoint = exports.Certificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.EventSubscription = null;
utilities.lazyLoad(exports, ["EventSubscription"], () => require("./eventSubscription"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.getReplicationInstance = null;
exports.getReplicationInstanceOutput = null;
utilities.lazyLoad(exports, ["getReplicationInstance", "getReplicationInstanceOutput"], () => require("./getReplicationInstance"));
exports.getReplicationSubnetGroup = null;
exports.getReplicationSubnetGroupOutput = null;
utilities.lazyLoad(exports, ["getReplicationSubnetGroup", "getReplicationSubnetGroupOutput"], () => require("./getReplicationSubnetGroup"));
exports.getReplicationTask = null;
exports.getReplicationTaskOutput = null;
utilities.lazyLoad(exports, ["getReplicationTask", "getReplicationTaskOutput"], () => require("./getReplicationTask"));
exports.ReplicationInstance = null;
utilities.lazyLoad(exports, ["ReplicationInstance"], () => require("./replicationInstance"));
exports.ReplicationSubnetGroup = null;
utilities.lazyLoad(exports, ["ReplicationSubnetGroup"], () => require("./replicationSubnetGroup"));
exports.ReplicationTask = null;
utilities.lazyLoad(exports, ["ReplicationTask"], () => require("./replicationTask"));
exports.S3Endpoint = null;
utilities.lazyLoad(exports, ["S3Endpoint"], () => require("./s3endpoint"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:dms/certificate:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "aws:dms/endpoint:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "aws:dms/eventSubscription:EventSubscription":
                return new exports.EventSubscription(name, undefined, { urn });
            case "aws:dms/replicationInstance:ReplicationInstance":
                return new exports.ReplicationInstance(name, undefined, { urn });
            case "aws:dms/replicationSubnetGroup:ReplicationSubnetGroup":
                return new exports.ReplicationSubnetGroup(name, undefined, { urn });
            case "aws:dms/replicationTask:ReplicationTask":
                return new exports.ReplicationTask(name, undefined, { urn });
            case "aws:dms/s3Endpoint:S3Endpoint":
                return new exports.S3Endpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "dms/certificate", _module);
pulumi.runtime.registerResourceModule("aws", "dms/endpoint", _module);
pulumi.runtime.registerResourceModule("aws", "dms/eventSubscription", _module);
pulumi.runtime.registerResourceModule("aws", "dms/replicationInstance", _module);
pulumi.runtime.registerResourceModule("aws", "dms/replicationSubnetGroup", _module);
pulumi.runtime.registerResourceModule("aws", "dms/replicationTask", _module);
pulumi.runtime.registerResourceModule("aws", "dms/s3Endpoint", _module);
