// @pulumi/aws/servicequotas/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceQuota = exports.getServiceQuotaOutput = exports.getServiceQuota = exports.getServiceOutput = exports.getService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceQuota = null;
exports.getServiceQuotaOutput = null;
utilities.lazyLoad(exports, ["getServiceQuota", "getServiceQuotaOutput"], () => require("./getServiceQuota"));
exports.ServiceQuota = null;
utilities.lazyLoad(exports, ["ServiceQuota"], () => require("./serviceQuota"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:servicequotas/serviceQuota:ServiceQuota":
                return new exports.ServiceQuota(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "servicequotas/serviceQuota", _module);
