// @pulumi/aws/quicksight/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.Template = exports.GroupMembership = exports.Group = exports.getQuicksightUserOutput = exports.getQuicksightUser = exports.getQuicksightGroupOutput = exports.getQuicksightGroup = exports.getDataSetOutput = exports.getDataSet = exports.Folder = exports.DataSource = exports.DataSet = exports.AccountSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccountSubscription = null;
utilities.lazyLoad(exports, ["AccountSubscription"], () => require("./accountSubscription"));
exports.DataSet = null;
utilities.lazyLoad(exports, ["DataSet"], () => require("./dataSet"));
exports.DataSource = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
exports.Folder = null;
utilities.lazyLoad(exports, ["Folder"], () => require("./folder"));
exports.getDataSet = null;
exports.getDataSetOutput = null;
utilities.lazyLoad(exports, ["getDataSet", "getDataSetOutput"], () => require("./getDataSet"));
exports.getQuicksightGroup = null;
exports.getQuicksightGroupOutput = null;
utilities.lazyLoad(exports, ["getQuicksightGroup", "getQuicksightGroupOutput"], () => require("./getQuicksightGroup"));
exports.getQuicksightUser = null;
exports.getQuicksightUserOutput = null;
utilities.lazyLoad(exports, ["getQuicksightUser", "getQuicksightUserOutput"], () => require("./getQuicksightUser"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.GroupMembership = null;
utilities.lazyLoad(exports, ["GroupMembership"], () => require("./groupMembership"));
exports.Template = null;
utilities.lazyLoad(exports, ["Template"], () => require("./template"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:quicksight/accountSubscription:AccountSubscription":
                return new exports.AccountSubscription(name, undefined, { urn });
            case "aws:quicksight/dataSet:DataSet":
                return new exports.DataSet(name, undefined, { urn });
            case "aws:quicksight/dataSource:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "aws:quicksight/folder:Folder":
                return new exports.Folder(name, undefined, { urn });
            case "aws:quicksight/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "aws:quicksight/groupMembership:GroupMembership":
                return new exports.GroupMembership(name, undefined, { urn });
            case "aws:quicksight/template:Template":
                return new exports.Template(name, undefined, { urn });
            case "aws:quicksight/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "quicksight/accountSubscription", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/dataSet", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/dataSource", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/folder", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/group", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/groupMembership", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/template", _module);
pulumi.runtime.registerResourceModule("aws", "quicksight/user", _module);
