// @pulumi/aws/ec2transitgateway/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcAttachmentAccepter = exports.VpcAttachment = exports.TransitGateway = exports.RouteTablePropagation = exports.RouteTableAssociation = exports.RouteTable = exports.Route = exports.PrefixListReference = exports.PolicyTableAssociation = exports.PolicyTable = exports.PeeringAttachmentAccepter = exports.PeeringAttachment = exports.MulticastGroupSource = exports.MulticastGroupMember = exports.MulticastDomainAssociation = exports.MulticastDomain = exports.InstanceState = exports.getVpnAttachmentOutput = exports.getVpnAttachment = exports.getVpcAttachmentsOutput = exports.getVpcAttachments = exports.getVpcAttachmentOutput = exports.getVpcAttachment = exports.getTransitGatewayOutput = exports.getTransitGateway = exports.getRouteTablePropagationsOutput = exports.getRouteTablePropagations = exports.getRouteTableAssociationsOutput = exports.getRouteTableAssociations = exports.getRouteTableOutput = exports.getRouteTable = exports.getPeeringAttachmentOutput = exports.getPeeringAttachment = exports.getMulticastDomainOutput = exports.getMulticastDomain = exports.getDirectConnectGatewayAttachmentOutput = exports.getDirectConnectGatewayAttachment = exports.getConnectPeerOutput = exports.getConnectPeer = exports.getConnectOutput = exports.getConnect = exports.getAttachmentsOutput = exports.getAttachments = exports.getAttachmentOutput = exports.getAttachment = exports.ConnectPeer = exports.Connect = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connect = null;
utilities.lazyLoad(exports, ["Connect"], () => require("./connect"));
exports.ConnectPeer = null;
utilities.lazyLoad(exports, ["ConnectPeer"], () => require("./connectPeer"));
exports.getAttachment = null;
exports.getAttachmentOutput = null;
utilities.lazyLoad(exports, ["getAttachment", "getAttachmentOutput"], () => require("./getAttachment"));
exports.getAttachments = null;
exports.getAttachmentsOutput = null;
utilities.lazyLoad(exports, ["getAttachments", "getAttachmentsOutput"], () => require("./getAttachments"));
exports.getConnect = null;
exports.getConnectOutput = null;
utilities.lazyLoad(exports, ["getConnect", "getConnectOutput"], () => require("./getConnect"));
exports.getConnectPeer = null;
exports.getConnectPeerOutput = null;
utilities.lazyLoad(exports, ["getConnectPeer", "getConnectPeerOutput"], () => require("./getConnectPeer"));
exports.getDirectConnectGatewayAttachment = null;
exports.getDirectConnectGatewayAttachmentOutput = null;
utilities.lazyLoad(exports, ["getDirectConnectGatewayAttachment", "getDirectConnectGatewayAttachmentOutput"], () => require("./getDirectConnectGatewayAttachment"));
exports.getMulticastDomain = null;
exports.getMulticastDomainOutput = null;
utilities.lazyLoad(exports, ["getMulticastDomain", "getMulticastDomainOutput"], () => require("./getMulticastDomain"));
exports.getPeeringAttachment = null;
exports.getPeeringAttachmentOutput = null;
utilities.lazyLoad(exports, ["getPeeringAttachment", "getPeeringAttachmentOutput"], () => require("./getPeeringAttachment"));
exports.getRouteTable = null;
exports.getRouteTableOutput = null;
utilities.lazyLoad(exports, ["getRouteTable", "getRouteTableOutput"], () => require("./getRouteTable"));
exports.getRouteTableAssociations = null;
exports.getRouteTableAssociationsOutput = null;
utilities.lazyLoad(exports, ["getRouteTableAssociations", "getRouteTableAssociationsOutput"], () => require("./getRouteTableAssociations"));
exports.getRouteTablePropagations = null;
exports.getRouteTablePropagationsOutput = null;
utilities.lazyLoad(exports, ["getRouteTablePropagations", "getRouteTablePropagationsOutput"], () => require("./getRouteTablePropagations"));
exports.getTransitGateway = null;
exports.getTransitGatewayOutput = null;
utilities.lazyLoad(exports, ["getTransitGateway", "getTransitGatewayOutput"], () => require("./getTransitGateway"));
exports.getVpcAttachment = null;
exports.getVpcAttachmentOutput = null;
utilities.lazyLoad(exports, ["getVpcAttachment", "getVpcAttachmentOutput"], () => require("./getVpcAttachment"));
exports.getVpcAttachments = null;
exports.getVpcAttachmentsOutput = null;
utilities.lazyLoad(exports, ["getVpcAttachments", "getVpcAttachmentsOutput"], () => require("./getVpcAttachments"));
exports.getVpnAttachment = null;
exports.getVpnAttachmentOutput = null;
utilities.lazyLoad(exports, ["getVpnAttachment", "getVpnAttachmentOutput"], () => require("./getVpnAttachment"));
exports.InstanceState = null;
utilities.lazyLoad(exports, ["InstanceState"], () => require("./instanceState"));
exports.MulticastDomain = null;
utilities.lazyLoad(exports, ["MulticastDomain"], () => require("./multicastDomain"));
exports.MulticastDomainAssociation = null;
utilities.lazyLoad(exports, ["MulticastDomainAssociation"], () => require("./multicastDomainAssociation"));
exports.MulticastGroupMember = null;
utilities.lazyLoad(exports, ["MulticastGroupMember"], () => require("./multicastGroupMember"));
exports.MulticastGroupSource = null;
utilities.lazyLoad(exports, ["MulticastGroupSource"], () => require("./multicastGroupSource"));
exports.PeeringAttachment = null;
utilities.lazyLoad(exports, ["PeeringAttachment"], () => require("./peeringAttachment"));
exports.PeeringAttachmentAccepter = null;
utilities.lazyLoad(exports, ["PeeringAttachmentAccepter"], () => require("./peeringAttachmentAccepter"));
exports.PolicyTable = null;
utilities.lazyLoad(exports, ["PolicyTable"], () => require("./policyTable"));
exports.PolicyTableAssociation = null;
utilities.lazyLoad(exports, ["PolicyTableAssociation"], () => require("./policyTableAssociation"));
exports.PrefixListReference = null;
utilities.lazyLoad(exports, ["PrefixListReference"], () => require("./prefixListReference"));
exports.Route = null;
utilities.lazyLoad(exports, ["Route"], () => require("./route"));
exports.RouteTable = null;
utilities.lazyLoad(exports, ["RouteTable"], () => require("./routeTable"));
exports.RouteTableAssociation = null;
utilities.lazyLoad(exports, ["RouteTableAssociation"], () => require("./routeTableAssociation"));
exports.RouteTablePropagation = null;
utilities.lazyLoad(exports, ["RouteTablePropagation"], () => require("./routeTablePropagation"));
exports.TransitGateway = null;
utilities.lazyLoad(exports, ["TransitGateway"], () => require("./transitGateway"));
exports.VpcAttachment = null;
utilities.lazyLoad(exports, ["VpcAttachment"], () => require("./vpcAttachment"));
exports.VpcAttachmentAccepter = null;
utilities.lazyLoad(exports, ["VpcAttachmentAccepter"], () => require("./vpcAttachmentAccepter"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ec2transitgateway/connect:Connect":
                return new exports.Connect(name, undefined, { urn });
            case "aws:ec2transitgateway/connectPeer:ConnectPeer":
                return new exports.ConnectPeer(name, undefined, { urn });
            case "aws:ec2transitgateway/instanceState:InstanceState":
                return new exports.InstanceState(name, undefined, { urn });
            case "aws:ec2transitgateway/multicastDomain:MulticastDomain":
                return new exports.MulticastDomain(name, undefined, { urn });
            case "aws:ec2transitgateway/multicastDomainAssociation:MulticastDomainAssociation":
                return new exports.MulticastDomainAssociation(name, undefined, { urn });
            case "aws:ec2transitgateway/multicastGroupMember:MulticastGroupMember":
                return new exports.MulticastGroupMember(name, undefined, { urn });
            case "aws:ec2transitgateway/multicastGroupSource:MulticastGroupSource":
                return new exports.MulticastGroupSource(name, undefined, { urn });
            case "aws:ec2transitgateway/peeringAttachment:PeeringAttachment":
                return new exports.PeeringAttachment(name, undefined, { urn });
            case "aws:ec2transitgateway/peeringAttachmentAccepter:PeeringAttachmentAccepter":
                return new exports.PeeringAttachmentAccepter(name, undefined, { urn });
            case "aws:ec2transitgateway/policyTable:PolicyTable":
                return new exports.PolicyTable(name, undefined, { urn });
            case "aws:ec2transitgateway/policyTableAssociation:PolicyTableAssociation":
                return new exports.PolicyTableAssociation(name, undefined, { urn });
            case "aws:ec2transitgateway/prefixListReference:PrefixListReference":
                return new exports.PrefixListReference(name, undefined, { urn });
            case "aws:ec2transitgateway/route:Route":
                return new exports.Route(name, undefined, { urn });
            case "aws:ec2transitgateway/routeTable:RouteTable":
                return new exports.RouteTable(name, undefined, { urn });
            case "aws:ec2transitgateway/routeTableAssociation:RouteTableAssociation":
                return new exports.RouteTableAssociation(name, undefined, { urn });
            case "aws:ec2transitgateway/routeTablePropagation:RouteTablePropagation":
                return new exports.RouteTablePropagation(name, undefined, { urn });
            case "aws:ec2transitgateway/transitGateway:TransitGateway":
                return new exports.TransitGateway(name, undefined, { urn });
            case "aws:ec2transitgateway/vpcAttachment:VpcAttachment":
                return new exports.VpcAttachment(name, undefined, { urn });
            case "aws:ec2transitgateway/vpcAttachmentAccepter:VpcAttachmentAccepter":
                return new exports.VpcAttachmentAccepter(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/connect", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/connectPeer", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/instanceState", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/multicastDomain", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/multicastDomainAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/multicastGroupMember", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/multicastGroupSource", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/peeringAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/peeringAttachmentAccepter", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/policyTable", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/policyTableAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/prefixListReference", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/route", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/routeTable", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/routeTableAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/routeTablePropagation", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/transitGateway", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/vpcAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "ec2transitgateway/vpcAttachmentAccepter", _module);
